<!DOCTYPE html>
<html>
<head>
<title>comet</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link rel="stylesheet" type="text/css" href="../styles.css">
<script type="text/javascript"
  src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/javascript">
  MathJax.Hub.Config({jax: ['input/TeX','output/HTML-CSS'], displayAlign: 'left'});
</script>
<script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-26136956-1']);
  _gaq.push(['_trackPageview']);
  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>
<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.8.3/jquery.min.js"></script>
<script type="text/javascript">
  // Main Menu
  $( document ).ready(function() {
      var pull 		= $('.btn');
        menu 		= $('nav ul');
        menuHeight	= menu.height();
      $(pull).on('click', function(e) {
        e.preventDefault();
        menu.slideToggle();
      });
      $(window).resize(function(){
          var w = $(window).width();
          if(w > 320 && menu.is(':hidden')) {
            menu.removeAttr('style');
          } 
      });
  });
</script>
</head>
<body>
  <div class="page-wrap">
    <nav>
      <div class="btn">
        </div>
        <img src="../images/crux-logo.png" id="logo"></a>
      <ul id="navitems">
          <li><a href="../index.html">Home</a></li>
          <li><a href="../download.html">Download</a></li>
          <li><a href="../fileformats.html">File Formats</a></li>
            <li><a href="http://groups.google.com/group/crux-users">Contact</a></li> <!--Link to google support board-->
        </ul>
    </nav>
    <div id="content" class="autogenerated">
    <!-- START CONTENT -->
<h1>comet</h1>
<h2>Usage:</h2>
<p><code>crux comet [options] &lt;input spectra&gt; &lt;database_name&gt;</code></p>
<h2>Description:</h2>
<p>This command searches a protein database with a set of spectra, assigning peptide sequences to the observed spectra. This search engine was developed by Jimmy Eng at the University of Washington Proteomics Resource.</p><p>Although its history goes back two decades, the Comet search engine was first made publicly available in August 2012 on SourceForge. Comet is multithreaded and supports multiple input and output formats.</p><blockquote><a href="http://onlinelibrary.wiley.com/doi/10.1002/pmic.201200439/abstract">&quot;Comet: an open source tandem mass spectrometry sequence database search tool.&quot;</a> Eng JK, Jahan TA, Hoopmann MR. <em>Proteomics</em>. 2012 Nov 12. doi: 10.1002/pmic201200439</blockquote>
<h2>Input:</h2>
<ul>
  <li><code>input spectra</code> &ndash; The name of one or more files from which to parse the spectra. Valid formats include mzXML, mzML, mz5, raw, ms2, and cms2. Files in mzML or mzXML may be compressed with gzip. RAW files can be parsed only under windows and if the appropriate libraries were included at compile time. Multiple files can be included on the command line (space delimited), prior to the name of the database.</li>
  <li><code>database_name</code> &ndash; A full or relative path to the sequence database, in FASTA or PEFF format, to search. Example databases include RefSeq or UniProt.  The database can contain amino acid sequences or nucleic acid sequences. If sequences are amino acid sequences, set the parameter "nucleotide_reading_frame = 0". If the sequences are nucleic acid sequences, you must instruct Comet to translate these to amino acid sequences. Do this by setting nucleotide_reading_frame" to a value between 1 and 9.</li>
</ul>
<h2>Output:</h2>
<p>The program writes files to the folder <code>crux-output</code> by default. The name of the output folder can be set by the user using the <code>--output-dir</code> option. The following files will be created:
<ul>
  <li><code>comet.target.txt</code> &ndash; a tab-delimited text file containing the target PSMs. See <a href="../file-formats/txt-format.html">txt file format</a> for a list of the fields.</li>
  <li><code>comet.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
  <li><code>comet.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to standard error.</li>
</ul>
<h2>Options:</h2>
<ul style="list-style-type: none;">
<li class="nobullet">
<h3>Database</h3>
<ul>
  <li class="nobullet"><code>--decoy_search &lt;integer&gt;</code> &ndash; 0=no, 1=concatenated search, 2=separate search. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--peff_format &lt;integer&gt;</code> &ndash; 0=normal FASTA format,1=PEFF PSI-MOD modifications and amino acid variants,2=PEFF Unimod modifications and amino acid variants,3=PEFF PSI-MOD modifications, skipping amino acid variants,4=PEFF Unimod modifications, skipping amino acid variants,5=PEFF amino acid variants, skipping PEFF modifications. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--peff_obo &lt;string&gt;</code> &ndash; A full or relative path to the OBO file used with a PEFF search. Supported OBO formats are PSI-Mod and Unimod OBO files. Which OBO file you use depends on your PEFF input file. This parameter is ignored if "peff_format = 0". There is no default value if this parameter is missing. Default = <code>&lt;empty&gt;</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>CPU threads</h3>
<ul>
  <li class="nobullet"><code>--num_threads &lt;integer&gt;</code> &ndash; 0=poll CPU to set num threads; else specify num threads directly. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Masses</h3>
<ul>
  <li class="nobullet"><code>--peptide_mass_tolerance &lt;float&gt;</code> &ndash; Controls the mass tolerance value.  The mass tolerance is set at +/- the specified number i.e. an entered value of "1.0" applies a -1.0 to +1.0 tolerance. The units of the mass tolerance is controlled by the parameter "peptide_mass_units".  Default = <code>3</code>.</li>
  <li class="nobullet"><code>--auto_peptide_mass_tolerance false|warn|fail</code> &ndash; Automatically estimate optimal value for the peptide_mass_tolerance parameter from the spectra themselves. false=no estimation, warn=try to estimate but use the default value in case of failure, fail=try to estimate and quit in case of failure. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--peptide_mass_units &lt;integer&gt;</code> &ndash; 0=amu, 1=mmu, 2=ppm. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--mass_type_parent &lt;integer&gt;</code> &ndash; 0=average masses, 1=monoisotopic masses. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--mass_type_fragment &lt;integer&gt;</code> &ndash; 0=average masses, 1=monoisotopic masses. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--precursor_tolerance_type &lt;integer&gt;</code> &ndash; 0=singly charged peptide mass, 1=precursor m/z. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--isotope_error &lt;integer&gt;</code> &ndash; 0=off, 1=0/1 (C13 error), 2=0/1/2, 3=0/1/2/3, 4=--8/-4/0/4/8 (for +4/+8 labeling), 5=-1/0/1/2/3. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Search enzyme</h3>
<ul>
  <li class="nobullet"><code>--search_enzyme_number &lt;integer&gt;</code> &ndash; Specify a search enzyme from the end of the parameter file. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--search_enzyme2_number &lt;integer&gt;</code> &ndash; Specify a second search enzyme from the end of the parameter file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--num_enzyme_termini &lt;integer&gt;</code> &ndash; valid values are 1 (semi-digested), 2 (fully digested), 8 N-term, 9 C-term. Default = <code>2</code>.</li>
  <li class="nobullet"><code>--allowed_missed_cleavage &lt;integer&gt;</code> &ndash; Maximum value is 5; for enzyme search. Default = <code>2</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Fragment ions</h3>
<ul>
  <li class="nobullet"><code>--fragment_bin_tol &lt;float&gt;</code> &ndash; Binning to use on fragment ions. Default = <code>1.000507</code>.</li>
  <li class="nobullet"><code>--fragment_bin_offset &lt;float&gt;</code> &ndash; Offset position to start the binning (0.0 to 1.0). Default = <code>0.4</code>.</li>
  <li class="nobullet"><code>--auto_fragment_bin_tol false|warn|fail</code> &ndash; Automatically estimate optimal value for the fragment_bin_tol parameter from the spectra themselves. false=no estimation, warn=try to estimate but use the default value in case of failure, fail=try to estimate and quit in case of failure. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--theoretical_fragment_ions &lt;integer&gt;</code> &ndash; 0=default peak shape, 1=M peak only. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--use_A_ions &lt;integer&gt;</code> &ndash; Controls whether or not A-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use_B_ions &lt;integer&gt;</code> &ndash; Controls whether or not B-ions are considered in the search (0 - no, 1 - yes). Default = <code>1</code>.</li>
  <li class="nobullet"><code>--use_C_ions &lt;integer&gt;</code> &ndash; Controls whether or not C-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use_X_ions &lt;integer&gt;</code> &ndash; Controls whether or not X-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use_Y_ions &lt;integer&gt;</code> &ndash; Controls whether or not Y-ions are considered in the search (0 - no, 1 - yes). Default = <code>1</code>.</li>
  <li class="nobullet"><code>--use_Z_ions &lt;integer&gt;</code> &ndash; Controls whether or not Z-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use_Z1_ions &lt;integer&gt;</code> &ndash; Controls whether or not Z1-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use_NL_ions &lt;integer&gt;</code> &ndash; 0=no, 1= yes to consider NH3/H2O neutral loss peak. Default = <code>1</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>mzXML/mzML parameters</h3>
<ul>
  <li class="nobullet"><code>--scan_range &lt;string&gt;</code> &ndash; Start and scan scan range to search; 0 as first entry ignores parameter. Default = <code>0 0</code>.</li>
  <li class="nobullet"><code>--precursor_charge &lt;string&gt;</code> &ndash; Precursor charge range to analyze; does not override mzXML charge; 0 as first entry ignores parameter. Default = <code>0 0</code>.</li>
  <li class="nobullet"><code>--override_charge &lt;integer&gt;</code> &ndash; Specifies the whether to override existing precursor charge state information when present in the files with the charge range specified by the "precursor_charge" parameter. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--ms_level &lt;integer&gt;</code> &ndash; MS level to analyze, valid are levels 2 or 3. Default = <code>2</code>.</li>
  <li class="nobullet"><code>--activation_method ALL|CID|ECD|ETD+SA|ETD|PQD|HCD|IRMPD</code> &ndash; Specifies which scan types are searched. Default = <code>ALL</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Miscellaneous parameters</h3>
<ul>
  <li class="nobullet"><code>--clip_nterm_methionine &lt;integer&gt;</code> &ndash; 0=leave sequences as-is; 1=also consider sequence w/o N-term methionine. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--decoy_prefix &lt;string&gt;</code> &ndash; Specifies the prefix of the protein names that indicates a decoy. Default = <code>decoy_</code>.</li>
  <li class="nobullet"><code>--digest_mass_range &lt;string&gt;</code> &ndash; MH+ peptide mass range to analyze. Default = <code>600.0 5000.0</code>.</li>
  <li class="nobullet"><code>--equal_I_and_L &lt;integer&gt;</code> &ndash; This parameter controls whether the Comet treats isoleucine (I) and leucine (L) as the same/equivalent with respect to a peptide identification. 0 treats I and L as different, 1 treats I and L as the same. The default value is "1" Default = <code>1</code>.</li>
  <li class="nobullet"><code>--mass_offsets &lt;string&gt;</code> &ndash; Specifies one or more mass offsets to apply. This value(s) are effectively subtracted from each precursor mass such that peptides that are smaller than the precursor mass by the offset value can still be matched to the respective spectrum. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--max_duplicate_proteins &lt;integer&gt;</code> &ndash; defines the maximum number of proteins (identifiers/accessions) to report. If a peptide is present in 6 total protein sequences, there is one (first) reference protein and 5 additional duplicate proteins. This parameter controls how many of those 5 additional duplicate proteins are reported.If "decoy_search = 2" is set to report separate target and decoy results, this parameter will be applied to the target and decoy outputs separately. If set to "-1", there will be no limit on the number of reported additional proteins. The default value is "20" if this parameter is missing. Default = <code>20</code>.</li>
  <li class="nobullet"><code>--max_fragment_charge &lt;integer&gt;</code> &ndash; Set maximum fragment charge state to analyze (allowed max 5). Default = <code>3</code>.</li>
  <li class="nobullet"><code>--max_index_runtime &lt;integer&gt;</code> &ndash; Sets the maximum indexed database search run time for a scan/query. Valid values are integers 0 or higher representing the maximum run time in milliseconds. As Comet loops through analyzing peptides from the database index file, it checks the cummulative run time of that spectrum search after each peptide is analyzed. If the run time exceeds the value set for this parameter, the search is aborted and the best peptide result analyzed up to that point is returned. To have no maximum search time, set this parameter value to "0". The default value is "0". Default = <code>0</code>.</li>
  <li class="nobullet"><code>--max_precursor_charge &lt;integer&gt;</code> &ndash; Set maximum precursor charge state to analyze (allowed max 9). Default = <code>6</code>.</li>
  <li class="nobullet"><code>--num_results &lt;integer&gt;</code> &ndash; Number of search hits to store internally. Default = <code>50</code>.</li>
  <li class="nobullet"><code>--nucleotide_reading_frame &lt;integer&gt;</code> &ndash; 0=proteinDB, 1-6, 7=forward three, 8=reverse three, 9=all six. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--output_suffix &lt;string&gt;</code> &ndash; Specifies the suffix string that is appended to the base output name for the pep.xml, pin.xml, txt and sqt output files. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--peptide_length_range &lt;string&gt;</code> &ndash; Defines the length range of peptides to search. This parameter has two integer values. The first value is the minimum length cutoff and the second value is the maximum length cutoff. Only peptides within the specified length range are analyzed. The maximum peptide length that Comet can analyze is 63. The default values are "1 63". Default = <code>1 63</code>.</li>
  <li class="nobullet"><code>--precursor_NL_ions &lt;string&gt;</code> &ndash; Controls whether or not precursor neutral loss peaks are considered in the xcorr scoring. If left blank, this parameter is ignored.  To consider precursor neutral loss peaks, add one or more neutral loss mass value separated by a space.  Each entered mass value will be subtracted from the experimentral precursor mass and resulting neutral loss m/z values for all charge states (from 1 to precursor charge) will be analyzed. As these neutral loss peaks are analyzed along side fragment ion peaks, the fragment tolerance settings (fragment_bin_tol, fragment_bin_offset, theoretical_fragment_ion) apply to the precursor neutral loss peaks. The default value is blank/unused. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--skip_researching &lt;integer&gt;</code> &ndash; For '.out' file output only, 0=search everything again, 1=don't search if .out exists. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--spectrum_batch_size &lt;integer&gt;</code> &ndash; Maximum number of spectra to search at a time; 0 to search the entire scan range in one loop. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--text_file_extension &lt;string&gt;</code> &ndash; Specifies the a custom extension for output text file. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--explicit_deltacn &lt;integer&gt;</code> &ndash; 0=Comet deltaCn reported between the top peptide and the first dissimilar peptide, 1=Comet deltaCn reported between the top two peptides. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--old_mods_encoding &lt;integer&gt;</code> &ndash; 0=Comet will use mass based modification encodings, 1=Comet will use the old character based modification encodings. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Spectral processing</h3>
<ul>
  <li class="nobullet"><code>--minimum_peaks &lt;integer&gt;</code> &ndash; Minimum number of peaks in spectrum to search. Default = <code>10</code>.</li>
  <li class="nobullet"><code>--minimum_intensity &lt;float&gt;</code> &ndash; Minimum intensity value to read in. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--remove_precursor_peak &lt;integer&gt;</code> &ndash; 0=no, 1=yes, 2=all charge reduced precursor peaks (for ETD). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--remove_precursor_tolerance &lt;float&gt;</code> &ndash; +- Da tolerance for precursor removal. Default = <code>1.5</code>.</li>
  <li class="nobullet"><code>--clear_mz_range &lt;string&gt;</code> &ndash; For iTRAQ/TMT type data; will clear out all peaks in the specified m/z range. Default = <code>0.0 0.0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Variable modifications</h3>
<ul>
  <li class="nobullet"><code>--variable_mod01 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported. Each modification is specified using seven entries: "&lt;mass&gt; &lt;residues&gt; &lt;type&gt; &lt;max&gt; &lt;distance&gt; &lt;terminus&gt; &lt;force&gt;." Type is 0 for static mods and non-zero for variable mods. Note that that if you set the same type value on multiple modification entries, Comet will treat those variable modifications as a binary set. This means that all modifiable residues in the binary set must be unmodified or modified. Multiple binary sets can be specified by setting a different binary modification value. Max is an integer specifying the maximum number of modified residues possible in a peptide for this modification entry. Distance specifies the distance the modification is applied to from the respective terminus: -1 = no distance contraint; 0 = only applies to terminal residue; N = only applies to terminal residue through next N residues. Terminus specifies which terminus the distance constraint is applied to: 0 = protein N-terminus; 1 = protein C-terminus; 2 = peptide N-terminus; 3 = peptide C-terminus.Force specifies whether peptides must contain this modification: 0 = not forced to be present; 1 = modification is required. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod02 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod03 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod04 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod05 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod06 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod07 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod08 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod09 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--auto_modifications T|F</code> &ndash; Automatically infer modifications from the spectra themselves. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--max_variable_mods_in_peptide &lt;integer&gt;</code> &ndash; Specifies the total/maximum number of residues that can be modified in a peptide. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--require_variable_mod &lt;integer&gt;</code> &ndash; Controls whether the analyzed peptides must contain at least one variable modification. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Static modifications</h3>
<ul>
  <li class="nobullet"><code>--add_Cterm_peptide &lt;float&gt;</code> &ndash; Specifiy a static modification to the c-terminus of all peptides. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Nterm_peptide &lt;float&gt;</code> &ndash; Specify a static modification to the n-terminus of all peptides. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Cterm_protein &lt;float&gt;</code> &ndash; Specify a static modification to the c-terminal peptide of each protein. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Nterm_protein &lt;float&gt;</code> &ndash; Specify a static modification to the n-terminal peptide of each protein. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_A_alanine &lt;float&gt;</code> &ndash; Specify a static modification to the residue A. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_C_cysteine &lt;float&gt;</code> &ndash; Specify a static modification to the residue C. Default = <code>57.021464</code>.</li>
  <li class="nobullet"><code>--add_D_aspartic_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue D. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_E_glutamic_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue E. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_F_phenylalanine &lt;float&gt;</code> &ndash; Specify a static modification to the residue F. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_G_glycine &lt;float&gt;</code> &ndash; Specify a static modification to the residue G. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_H_histidine &lt;float&gt;</code> &ndash; Specify a static modification to the residue H. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_I_isoleucine &lt;float&gt;</code> &ndash; Specify a static modification to the residue I. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_K_lysine &lt;float&gt;</code> &ndash; Specify a static modification to the residue K. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_L_leucine &lt;float&gt;</code> &ndash; Specify a static modification to the residue L. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_M_methionine &lt;float&gt;</code> &ndash; Specify a static modification to the residue M. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_N_asparagine &lt;float&gt;</code> &ndash; Specify a static modification to the residue N. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_O_pyrrolysine &lt;float&gt;</code> &ndash; Specify a static modification to the residue O. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_P_proline &lt;float&gt;</code> &ndash; Specify a static modification to the residue P. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Q_glutamine &lt;float&gt;</code> &ndash; Specify a static modification to the residue Q. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_R_arginine &lt;float&gt;</code> &ndash; Specify a static modification to the residue R. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_S_serine &lt;float&gt;</code> &ndash; Specify a static modification to the residue S. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_T_threonine &lt;float&gt;</code> &ndash; Specify a static modification to the residue T. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_U_selenocysteine &lt;float&gt;</code> &ndash; Specify a static modification to the residue U. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_V_valine &lt;float&gt;</code> &ndash; Specify a static modification to the residue V. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_W_tryptophan &lt;float&gt;</code> &ndash; Specify a static modification to the residue W. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Y_tyrosine &lt;float&gt;</code> &ndash; Specify a static modification to the residue Y. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_B_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue B. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_J_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue J. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_X_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue X. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Z_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue Z. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>param-medic options</h3>
<ul>
  <li class="nobullet"><code>--pm-min-precursor-mz &lt;float&gt;</code> &ndash; Minimum precursor m/z value to use in measurement error estimation. Default = <code>400</code>.</li>
  <li class="nobullet"><code>--pm-max-precursor-mz &lt;float&gt;</code> &ndash; Minimum precursor m/z value to use in measurement error estimation. Default = <code>1800</code>.</li>
  <li class="nobullet"><code>--pm-min-frag-mz &lt;float&gt;</code> &ndash; Minimum fragment m/z value to use in measurement error estimation. Default = <code>150</code>.</li>
  <li class="nobullet"><code>--pm-max-frag-mz &lt;float&gt;</code> &ndash; Maximum fragment m/z value to use in measurement error estimation. Default = <code>1800</code>.</li>
  <li class="nobullet"><code>--pm-min-scan-frag-peaks &lt;integer&gt;</code> &ndash; Minimum fragment peaks an MS/MS scan must contain to be used in measurement error estimation. Default = <code>40</code>.</li>
  <li class="nobullet"><code>--pm-max-precursor-delta-ppm &lt;float&gt;</code> &ndash; Maximum ppm distance between precursor m/z values to consider two scans potentially generated by the same peptide for measurement error estimation. Default = <code>50</code>.</li>
  <li class="nobullet"><code>--pm-charges &lt;string&gt;</code> &ndash; Precursor charge states to consider MS/MS spectra from, in measurement error estimation, provided as comma-separated values. Default = <code>0,2,3,4</code>.</li>
  <li class="nobullet"><code>--pm-top-n-frag-peaks &lt;integer&gt;</code> &ndash; Number of most-intense fragment peaks to consider for measurement error estimation, per MS/MS spectrum. Default = <code>30</code>.</li>
  <li class="nobullet"><code>--pm-pair-top-n-frag-peaks &lt;integer&gt;</code> &ndash; Number of fragment peaks per spectrum pair to be used in fragment error estimation. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--pm-min-common-frag-peaks &lt;integer&gt;</code> &ndash; Number of the most-intense peaks that two spectra must share in order to potentially be generated by the same peptide, for measurement error estimation. Default = <code>20</code>.</li>
  <li class="nobullet"><code>--pm-max-scan-separation &lt;integer&gt;</code> &ndash; Maximum number of scans two spectra can be separated by in order to be considered potentially generated by the same peptide, for measurement error estimation. Default = <code>1000</code>.</li>
  <li class="nobullet"><code>--pm-min-peak-pairs &lt;integer&gt;</code> &ndash; Minimum number of peak pairs (for precursor or fragment) that must be successfully paired in order to attempt to estimate measurement error distribution. Default = <code>200</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Input and output</h3>
<ul>
  <li class="nobullet"><code>--fileroot &lt;string&gt;</code> &ndash; The fileroot string will be added as a prefix to all output file names. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--output-dir &lt;string&gt;</code> &ndash; The name of the directory where output files will be created. Default = <code>crux-output</code>.</li>
  <li class="nobullet"><code>--overwrite T|F</code> &ndash; Replace existing files if true or fail when trying to overwrite a file if false. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--parameter-file &lt;string&gt;</code> &ndash; A file containing parameters.  See the <a href="../file-formats/parameter-file.html">parameter documentation</a> page for details. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--verbosity &lt;integer&gt;</code> &ndash; Specify the verbosity of the current processes. Each level prints the following messages, including all those at lower verbosity levels: 0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on the progress of execution, 40-more progress information, 50-debug info, 60-detailed debug info. Default = <code>30</code>.</li>
  <li class="nobullet"><code>--output_mzidentmlfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write mzIdentML file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--output_pepxmlfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write pep.xml file. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--output_percolatorfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes write percolator file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--output_sqtfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write sqt file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--output_sqtstream &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write sqt file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--output_txtfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write tab-delimited text file. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--print_expect_score &lt;integer&gt;</code> &ndash; 0=no, 1=yes to replace Sp with expect in out & sqt. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--num_output_lines &lt;integer&gt;</code> &ndash; num peptide results to show. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--show_fragment_ions &lt;integer&gt;</code> &ndash; 0=no, 1=yes for out files only. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--sample_enzyme_number &lt;integer&gt;</code> &ndash; Sample enzyme which is possibly different than the one applied to the search. Used to calculate NTT & NMC in pepXML output. Default = <code>1</code>.</li>
</ul>
</li>

</ul>
    <!-- END CONTENT -->
    </div>
  </div>
<footer class="site-footer">
  <div id="centerfooter">
    <div class="footerimportantlinks">
      <img src="../images/linkicon.png" style="width:16px; height:16px"><h3>Important links</h3>
      <ul>
        <li><a href="../faq.html">Crux FAQ</a></li>
        <li><a href="../glossary.html">Glossary of terminology</a></li>
        <li><a href="http://scholar.google.com/citations?hl=en&user=Rw9S1HIAAAAJ">Google Scholar profile</a></li>
        <li><a href="https://github.com/crux-toolkit/crux-toolkit">GitHub</a></li>
        <li><a href="../release-notes.html">Release Notes</a></li>
        <li><a href="https://mailman1.u.washington.edu/mailman/listinfo/crux-users" title="Receive announcements of new versions">Join the mailing list</a></li>
        <li><a href="http://www.apache.org/licenses/LICENSE-2.0">Apache license</a></li>
        <li><a href="http://groups.google.com/group/crux-users">Support Board</a></li>
      </ul>
    </div>
    <div class="footerimportantlinks tutoriallinks">
      <img src="../images/tutorialicon.png" style="height:16px"><h3>Tutorials</h3>
      <ul>
        <li><a href="../tutorials/install.html">Installation</a></li>
        <li><a href="../tutorials/gettingstarted.html">Getting started with Crux</a></li>
        <li><a href="../tutorials/search.html">Running a simple search using Tide and Percolator</a></li>
        <li><a href="../tutorials/customizedsearch.html">Customization and search options</a></li>
        <li><a href="../tutorials/spectralcounts.html">Using spectral-counts</a></li>
      </ul>
    </div>
    <div id="footertext">
      <p>
        The original version of Crux was written by Chris Park and Aaron Klammer
        under the supervision
        of <a href="http://www.gs.washington.edu/faculty/maccoss.htm">Prof. Michael
        MacCoss</a>
        and <a href="http://noble.gs.washington.edu/~noble">Prof. William
        Stafford Noble</a> in the Department of Genome Sciences at the
        University of Washington, Seattle.  Website by <a href="http://www.yuvalboss.com/">Yuval Boss</a>. 
        <br />The complete list of contributors
        can be found <a href="../contributors.html">here</a>.
        <br />
        <br />
        Maintenance and development of Crux is funded by the <a href="https://www.nih.gov/">National Institutes of Health</a> award P41 GM103533. 
      </p>
    </div>
  </div>
</footer>
</body>
</html>