<!DOCTYPE html>
<html>
<head>
<title>diameter</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link rel="stylesheet" type="text/css" href="../styles.css">
<script>
  window.MathJax = {
    chtml: { displayAlign: 'left' },
    options: {
      ignoreHtmlClass: 'tex2jax_ignore',
      processHtmlClass: 'tex2jax_process'
    },
    tex: {
      autoload: {
        color: [],
        colorv2: ['color']
      },
      packages: {'[+]': ['noerrors']}
    },
    loader: {load: ['[tex]/noerrors']}
  };
</script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js" id="MathJax-script"></script>
<script type="text/javascript">
  MathJax.Hub.Config({jax: ['input/TeX','output/HTML-CSS'], displayAlign: 'left'});
</script>
<script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-26136956-1']);
  _gaq.push(['_trackPageview']);
  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.8.3/jquery.min.js"></script>
<script type="text/javascript">
  // Main Menu
  $( document ).ready(function() {
      var pull 		= $('.btn');
        menu 		= $('nav ul');
        menuHeight	= menu.height();
      $(pull).on('click', function(e) {
        e.preventDefault();
        menu.slideToggle();
      });
      $(window).resize(function(){
          var w = $(window).width();
          if(w > 320 && menu.is(':hidden')) {
            menu.removeAttr('style');
          } 
      });
  });
</script>
</head>
<body>
  <div class="page-wrap">
    <nav>
      <div class="btn">
        </div>
        <img src="../images/crux-logo.png" id="logo"></a>
      <ul id="navitems">
          <li><a href="../index.html">Home</a></li>
          <li><a href="../download.html">Download</a></li>
          <li><a href="../fileformats.html">File Formats</a></li>
            <li><a href="http://groups.google.com/group/crux-users">Contact</a></li> <!--Link to google support board-->
        </ul>
    </nav>
    <div id="content" class="autogenerated">
    <!-- START CONTENT -->
<h1>diameter</h1>
<h2>Usage:</h2>
<p><code>crux diameter [options] &lt;tide spectra file&gt; &lt;tide database&gt;</code></p>
<h2>Description:</h2>
<p>DIAmeter detects peptides from data-independent acquisition mass spectrometry data without requiring a spectral library. The input includes centroided DIA data and a proteome FASTA database. DIAmeter then searches the DIA data using Tide, allowing multiple peptide-spectrum matches (PSMs) per DIA spectrum. A subset of these PSMs are selected for further analysis, using a greedy bipartite graph matching algorithm. Finally, PSMs are augmented and filtered with auxiliary features describing various types of evidence supporting the detection of the associated peptide. The PSM feature vectors, the output of DIAmeter, should be processed subsequently by Percolator to induce a ranking on peptides. Percolator will assign each peptide a statistical confidence estimate, where highly ranked peptides are detected in the DIA data with stronger confidence. Further details are provided here:</p><blockquote>YY Lu, J Bilmes, RA Rodriguez-Mias, J Villen, and WS Noble. <a href="https://doi.org/10.1093/bioinformatics/btab284">&quot;DIAmeter: Matching peptides to data-independent acquisition mass spectrometry data&quot;</a>. <em>Bioinformatics</em>. 37(Supplement_1):i434–i442, 2021.</blockquote><p>DIAmeter performs several intermediate steps, as follows:</p><ol><li>If a FASTA file was provided, convert it to an index using tide-index. Otherwise, use the given Tide index.</li><li>Convert the given fragmentation spectra to a binary format.</li><li>Search the spectra against the database and extract the auxiliary features.</li><li>Store the results in Percolator input (PIN) format.</li><li>Run the PIN file through Percolator. </li></ol>
<h2>Input:</h2>
<ul>
  <li><code>tide spectra file</code> &ndash; The name of one or more files from which to parse the fragmentation spectra, in any of the file formats supported by ProteoWizard. Alternatively, the argument may be one or more binary spectrum files produced by a previous run of crux tide-search using the store-spectra parameter. Multiple files can be included on the command line (space delimited), prior to the name of the database.</li>
  <li><code>tide database</code> &ndash; Either a FASTA file or a directory containing a database index created by a previous run of crux tide-index.</li>
</ul>
<h2>Output:</h2>
<p>The program writes files to the folder <code>crux-output</code> by default. The name of the output folder can be set by the user using the <code>--output-dir</code> option. The following files will be created:
<ul>
  <li><code>diameter.psm-features.txt</code> &ndash; a tab-delimited text file containing the feature of the searched PSMs. </li>
  <li><code>diameter.psm-features.filtered.txt</code> &ndash; a tab-delimited text file containing the feature of the PSMs after filtering. </li>
  <li><code>diameter.features.pin</code> &ndash; the searched PSM results in Percolator input (PIN) format. </li>
  <li><code>diameter.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other Crux programs.</li>
  <li><code>diameter.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to the screen during execution.</li>
  <li><code>percolator-output</code> &ndash; the output of percolator after running the PIN file.</li>
</ul>
<h2>Options:</h2>
<ul style="list-style-type: none;">
<li class="nobullet">
<h3>diameter options</h3>
<ul>
  <li class="nobullet"><code>--predrt-files &lt;string&gt;</code> &ndash; The name of file from which to parse the predicted retention time of each peptide in the database. The file is tab-delimited where the first column is peptide and the second column is the predicted rt information. The rt prediction doesn't require normalization beforehand. If the peptide in the database is missing in the prediction, its predicted value will be imputed by the median of all predicted values. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--prec-ppm &lt;integer&gt;</code> &ndash; Tolerance used for matching precursors to spectra. Peptides must be within +/- ‘precursor-ppm’ parts-per-million (ppm) of the spectrum precursor m/z Default = <code>10</code>.</li>
  <li class="nobullet"><code>--frag-ppm &lt;integer&gt;</code> &ndash; Tolerance used for matching fragment ions to spectrum peaks. Fragment ions must be within +/- 'fragment-ppm' of the spectrum peak value. Default = <code>10</code>.</li>
  <li class="nobullet"><code>--diameter-instrument orbitrap|tof5600|tof6600|na</code> &ndash; Specify the instrument platform used to acquire the input spectra. This option selects among different sets of coefficient values for the scores computed by diameter. Specifically, the 'orbitrap' setting is equivalent to spectra-denoising=false, psm-filter=false, prec-ppm=10,frag-ppm=10; the 'tof5600' is equivalent to spectra-denoising=true, psm-filter=true, prec-ppm=30, frag-ppm=30, coeff-precursor=3.2, coeff-fragment=0.2, coeff-rtdiff=0.2, coeff-elution=0.2; the 'tof6600' is equivalent to spectra-denoising=false, psm-filter=true, prec-ppm=30, frag-ppm=30, coeff-precursor=25.6, coeff-fragment=0.2, coeff-rtdiff=0.2, coeff-elution=0. Use diameter-instrument=na to set individual parameters.  Otherwise, parameters set using diameter-instrument will override any parameters set separately.  Default = <code>na</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Precursor selection</h3>
<ul>
  <li class="nobullet"><code>--max-precursor-charge &lt;integer&gt;</code> &ndash; The maximum charge state of a spectra to consider in search. Default = <code>5</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Search parameters</h3>
<ul>
  <li class="nobullet"><code>--mz-bin-offset &lt;float&gt;</code> &ndash; In the discretization of the m/z axes of the observed and theoretical spectra, this parameter specifies the location of the left edge of the first bin, relative to mass = 0 (i.e., mz-bin-offset = 0.xx means the left edge of the first bin will be located at +0.xx Da). Default = <code>0.4</code>.</li>
  <li class="nobullet"><code>--mz-bin-width &lt;float&gt;</code> &ndash; Before calculation of the XCorr score, the m/z axes of the observed and theoretical spectra are discretized. This parameter specifies the size of each bin. The exact formula for computing the discretized m/z value is floor((x/mz-bin-width) + 1.0 - mz-bin-offset), where x is the observed m/z value. For low resolution ion trap ms/ms data 1.0005079 and for high resolution ms/ms 0.02 is recommended. Default = <code>0.02</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Input and output</h3>
<ul>
  <li class="nobullet"><code>--output-dir &lt;string&gt;</code> &ndash; The name of the directory where output files will be created. Default = <code>crux-output</code>.</li>
  <li class="nobullet"><code>--overwrite T|F</code> &ndash; Replace existing files if true or fail when trying to overwrite a file if false. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--top-match &lt;integer&gt;</code> &ndash; Specify the number of matches to report for each spectrum. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--verbosity &lt;integer&gt;</code> &ndash; Specify the verbosity of the current processes. Each level prints the following messages, including all those at lower verbosity levels: 0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on the progress of execution, 40-more progress information, 50-debug info, 60-detailed debug info. Default = <code>30</code>.</li>
</ul>
</li>

</ul>
    <!-- END CONTENT -->
    </div>
  </div>
<footer class="site-footer">
  <div id="centerfooter">
    <div class="footerimportantlinks">
      <img src="../images/linkicon.png" style="width:16px; height:16px"><h3>Important links</h3>
      <ul>
        <li><a href="../faq.html">Crux FAQ</a></li>
        <li><a href="../glossary.html">Glossary of terminology</a></li>
        <li><a href="http://scholar.google.com/citations?hl=en&user=Rw9S1HIAAAAJ">Google Scholar profile</a></li>
        <li><a href="https://github.com/crux-toolkit/crux-toolkit">GitHub</a></li>
        <li><a href="../release-notes.html">Release Notes</a></li>
        <li><a href="https://mailman1.u.washington.edu/mailman/listinfo/crux-users" title="Receive announcements of new versions">Join the mailing list</a></li>
        <li><a href="http://www.apache.org/licenses/LICENSE-2.0">Apache license</a></li>
        <li><a href="http://groups.google.com/group/crux-users">Support Board</a></li>
      </ul>
    </div>
    <div class="footerimportantlinks tutoriallinks">
      <img src="../images/tutorialicon.png" style="height:16px"><h3>Tutorials</h3>
      <ul>
        <li><a href="../tutorials/install.html">Installation</a></li>
        <li><a href="../tutorials/gettingstarted.html">Getting started with Crux</a></li>
        <li><a href="../tutorials/search.html">Running a simple search using Tide and Percolator</a></li>
        <li><a href="../tutorials/customizedsearch.html">Customization and search options</a></li>
        <li><a href="../tutorials/spectralcounts.html">Using spectral-counts</a></li>
      </ul>
    </div>
    <div id="footertext">
      <p>
        The original version of Crux was written by Chris Park and Aaron Klammer
        under the supervision
        of <a href="http://www.gs.washington.edu/faculty/maccoss.htm">Prof. Michael
        MacCoss</a>
        and <a href="http://noble.gs.washington.edu/~noble">Prof. William
        Stafford Noble</a> in the Department of Genome Sciences at the
        University of Washington, Seattle.  Website by <a href="http://www.yuvalboss.com/">Yuval Boss</a>. 
        <br />The complete list of contributors
        can be found <a href="../contributors.html">here</a>.
        <br />
        <br />
        Maintenance and development of Crux is funded by the <a href="https://www.nih.gov/">National Institutes of Health</a> award P41 GM103533. 
      </p>
    </div>
  </div>
</footer>
</body>
</html>