<!DOCTYPE html>
<html>
<head>
<title>kojak</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link rel="stylesheet" type="text/css" href="../styles.css">
<script>
  window.MathJax = {
    chtml: { displayAlign: 'left' },
    options: {
      ignoreHtmlClass: 'tex2jax_ignore',
      processHtmlClass: 'tex2jax_process'
    },
    tex: {
      autoload: {
        color: [],
        colorv2: ['color']
      },
      packages: {'[+]': ['noerrors']}
    },
    loader: {load: ['[tex]/noerrors']}
  };
</script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js" id="MathJax-script"></script>
<script type="text/javascript">
  MathJax.Hub.Config({jax: ['input/TeX','output/HTML-CSS'], displayAlign: 'left'});
</script>
<script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-26136956-1']);
  _gaq.push(['_trackPageview']);
  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.8.3/jquery.min.js"></script>
<script type="text/javascript">
  // Main Menu
  $( document ).ready(function() {
      var pull 		= $('.btn');
        menu 		= $('nav ul');
        menuHeight	= menu.height();
      $(pull).on('click', function(e) {
        e.preventDefault();
        menu.slideToggle();
      });
      $(window).resize(function(){
          var w = $(window).width();
          if(w > 320 && menu.is(':hidden')) {
            menu.removeAttr('style');
          } 
      });
  });
</script>
</head>
<body>
  <div class="page-wrap">
    <nav>
      <div class="btn">
        </div>
        <img src="../images/crux-logo.png" id="logo"></a>
      <ul id="navitems">
          <li><a href="../index.html">Home</a></li>
          <li><a href="../download.html">Download</a></li>
          <li><a href="../fileformats.html">File Formats</a></li>
            <li><a href="http://groups.google.com/group/crux-users">Contact</a></li> <!--Link to google support board-->
        </ul>
    </nav>
    <div id="content" class="autogenerated">
    <!-- START CONTENT -->
<h1>kojak</h1>
<h2>Usage:</h2>
<p><code>crux kojak [options] &lt;input spectra&gt; &lt;protein database&gt;</code></p>
<h2>Description:</h2>
<p>Kojak is a database search tool for the identification of cross-linked peptides from mass spectra. This search engine was developed and is maintained by Michael Hoopmann at the Institute for Systems Biology. Additional information about Kojak can be found at <a href="http://www.kojak-ms.org">kojak-ms.org</a>.</p><p>If you use Kojak, please cite:</p> <p><a href="https://doi.org/10.1021/pr501321h">"Kojak: Efficient analysis of chemically cross-linked protein complexes."</a> Hoopmann MR, Zelter A, Johnson RS, Riffle M, MacCoss MJ, Davis TN, Moritz RL. <em>J Proteome Res</em> 2015 May 1. doi: 10.1021/pr501321h
<h2>Input:</h2>
<ul>
  <li><code>input spectra</code> &ndash; The name of one or more files from which to parse the spectra. Valid formats include mzXML, mzML, mz5, raw, ms2, and cms2. Files in mzML or mzXML may be compressed with gzip. RAW files can be parsed only under windows and if the appropriate libraries were included at compile time. Multiple files can be included on the command line (space delimited), prior to the name of the database.</li>
  <li><code>protein database</code> &ndash; The name of the fasta file containing the amino acid protein sequences to be searched. Kojak can generate decoy sequences internally, or they may be in this file (see the <code>decoy_filter</code> option for details). It is recommended to include the full path in the name of the file.</li>
</ul>
<h2>Output:</h2>
<p>The program writes files to the folder <code>crux-output</code> by default. The name of the output folder can be set by the user using the <code>--output-dir</code> option. The following files will be created:
<ul>
  <li><code>*.kojak.txt</code> &ndash; a tab-delimited text file containing the PSMs. See the <a href="../file-formats/txt-format.html">txt file format</a> for a list of the fields. The "*" indicates the root of the input spectrum file(s).</li>
  <li><code>*.log</code> &ndash; a log file containing specific information pertaining to the analysis of the corresponding spectrum file.</li>
  <li><code>kojak.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
  <li><code>kojak.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to standard error. These are mainly crux-specific messages.</li>
</ul>
<h2>Options:</h2>
<ul style="list-style-type: none;">
<li class="nobullet">
<h3>kojak options</h3>
<ul>
  <li class="nobullet"><code>--min_peptide_score &lt;float&gt;</code> &ndash; The minimum peptide score threshold for the first (alpha) peptide during crosslink analysis. During the first pass in the analysis, if the top scoring alpha peptides do not exceed this threshold, they will not be considered for pairing with a second (beta) peptide during the second pass of the analysis.  Default = <code>0.1</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Amino acid modifications</h3>
<ul>
  <li class="nobullet"><code>--fixed_modification &lt;string&gt;</code> &ndash; Specifies a mass adjustment to be applied to all indicated amino acids prior to spectral analysis. Amino acids are identified by their single letter designation. N-terminal and C-terminal fixed modifications are designated by n and c, respectively. The relative mass difference, positive or negative, is listed after the amino acid, separated by a space. If multiple fixed modification masses are desired, provide them as a comma-separated list enclosed in quotation marks. For example: "C 57.02146, nK 42.01057". Default = <code>C 57.02146</code>.</li>
  <li class="nobullet"><code>--fixed_modification_protC &lt;float&gt;</code> &ndash; Specifies a mass adjustment to be applied to all protein C-termini prior to spectral analysis. The relative mass difference may be any non-zero number. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fixed_modification_protN &lt;float&gt;</code> &ndash; Specifies a mass adjustment to be applied to all protein N-termini prior to spectral analysis. The relative mass difference may be any non-zero number. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--modification &lt;string&gt;</code> &ndash; Specifies a dynamic mass adjustment to be applied to all indicated amino acids during spectral analysis. Peptides containing the indicated amino acids are tested with and without the dynamic modification mass. Amino acids are identified by their single letter designation. N-terminal and C-terminal dynamic peptide modifications are designated by n and c, respectively. The relative mass difference, positive or negative, is listed after the amino acid, separated by a space. If multiple dynamic modification masses are desired, including to the same amino acid, provide them as a comma-separated list enclosed with quotation marks. For example: "M 15.9949, STY 79.966331". Default = <code>M 15.9949</code>.</li>
  <li class="nobullet"><code>--modification_protC &lt;string&gt;</code> &ndash; Specifies a dynamic mass adjustment to be applied to protein C-terminal amino acids during spectral analysis. Peptides containing the protein C-terminus are tested with and without the dynamic modification mass. The relative mass difference can be any non-zero value. If multiple dynamic protein C-terminal modification masses are desired, provide them as a comma-separated list enclosed in quotation marks. For example, "56.037448, -58.005479". Default = <code>0</code>.</li>
  <li class="nobullet"><code>--modification_protN &lt;string&gt;</code> &ndash; Specifies a dynamic mass adjustment to be applied to protein N-terminal amino acids during spectral analysis. Peptides containing the protein N-terminus are tested with and without the dynamic modification mass. The N-terminus includes both the leading and 2nd amino acid, in case of removal of the leading amino acid. The relative mass difference can be any non-zero value. If multiple dynamic protein N-terminal modification masses are desired, provide them as a comma-separated list enclosed in quotation marks. For example, "42.01055, 0.984016". Default = <code>0</code>.</li>
  <li class="nobullet"><code>--max_mods_per_peptide &lt;integer&gt;</code> &ndash; Indicates the maximum number of differential mass modifications allowed for a peptide sequence. Default = <code>2</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Enzymatic digestion</h3>
<ul>
  <li class="nobullet"><code>--max_miscleavages &lt;integer&gt;</code> &ndash; Number of missed enzyme cleavages allowed. If your digestion enzyme cuts at the same amino acids involved in cross-linking, then this number must be greater than 0 to identify linked peptides. In such cases, a minimum value of 2 is required to identify loop-links. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--kojak_enzyme &lt;string&gt;</code> &ndash; An enzyme string code is used to define amino acid cut sites when parsing protein sequences. Following the code, a separate label can be used to name the enzyme used. The rules for peptide parsing are similar to other database search engines such as X!Tandem: 1) cleavage amino acids are specified in square braces: [], 2) a vertical line, |, indicates N- or C-terminal to the residue, 3) exception amino acids are specified in curly braces: {}. Default = <code>[KR] Trypsin</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Search parameters</h3>
<ul>
  <li class="nobullet"><code>--top_count &lt;integer&gt;</code> &ndash; This parameter specifies the number of top scoring peptides to store in the first pass of the Kojak analysis. A second pass follows, pairingcross-linked peptides to these top sequences to produce the final cross-linked peptide score. Setting this number too low will cause cross-linked sequences to be missed. Setting this number too high will degrade the performance of the algorithm. Optimal settings will depend on database size and the number of modifications in the search. Recommended values are between 5 and 50 (20 is probably a good start). Default = <code>20</code>.</li>
  <li class="nobullet"><code>--e_value_depth &lt;integer&gt;</code> &ndash; Specifies the minimum number of tests to be present in the histogram for e-value calculations. A larger number better resolves the histogram and improves the e-value estimation for the peptide sequences in each spectrum. However, larger numbers also increase computation time. The recommended values are between 2000 and 10000 Default = <code>5000</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Cross-linking parameters</h3>
<ul>
  <li class="nobullet"><code>--cross_link &lt;string&gt;</code> &ndash; Specifies the sites of cross-linking and mass modification. Four values specify a cross-link. The first two values are one or more amino acid letters (uppercase only) that can be linked. These can be the same or different depending on whether the cross-linker is homobifunctional or heterobifunctional. Use lowercase ‘n’ or ‘c’ if the linker can bind the protein termini. The third value is the net mass value of the cross-linker when bound to the peptides. The mass can be any real number, positive or negative. The identifier is any name desired for the cross-linker. If the data contain multiple cross-linkers, provide them as a comma-separated list enclosed with quotation marks. For example, a sample cross-linked with both BS3 and EDC could be specified as "nK nK 138.068074 BS3, DE nK -18.0106 EDC". Default = <code>nK nK 138.068074 BS3</code>.</li>
  <li class="nobullet"><code>--mono_link &lt;string&gt;</code> &ndash; Specifies the sites of incomplete cross-linking (i.e. a mono-link) and mass modification. Two values follow this parameter, separated by spaces. The first value is one or more amino acid letters (uppercase only) that can be linked. Use lowercase ‘n’ or ‘c’ if the linker can bind the protein termini. The second value is the net mass of the incomplete cross-link reaction. The mass can be any real number, positive or negative. If multiple mono-links are possible (e.g. with a heterobifunctional cross-linker), provide them as a comma-separated list enclosed in quotation marks. For example: "nK 156.0786, nK 155.0946". Default = <code>nK 156.0786</code>.</li>
  <li class="nobullet"><code>--diff_mods_on_xl T|F</code> &ndash; Searching for differential modifications increases search times exponentially. This increase in computation can be exacerbated when searching for differential modifications on cross-linked peptides. Such computation can be avoided if is known that the cross-linked peptides should not have differential modifications. In these cases, this setting can be turned off. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--mono_links_on_xl T|F</code> &ndash; When multiple sites of linkage are available on a peptide, it is possible for that peptide to be linked to a second peptide at one site and contain a mono-link at another site. If such instances are considered rare due to the experimental conditions, then this parameter can be disabled to improve computation time. Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Database</h3>
<ul>
  <li class="nobullet"><code>--decoy_filter &lt;string&gt;</code> &ndash; This parameter requires two values. The first value is a short, case-sensitive string of characters that appears in the name of every decoy protein sequence in the database. The second value is either 0 or 1, where 0 indicates that these decoy sequences are already provided in the FASTA database supplied by the user, and 1 indicates Kojak should automatically generate the decoy sequences and preface the protein names with the characters supplied in the first value. If Kojak is requested to generate decoy sequences, it will save the full complement of target and decoy sequences as a fasta file in the output directory at the end of analysis. Kojak generates decoy sequences by reversing the amino acids between enzymatic cleavage sites in the protein sequence. The sites of enzymatic cleavage are determined by the rules supplied with the <code>kojak_enzyme</code> parameter. The leading methionine in the sequences, however, remains fixed. This approach ensures that, with very few exceptions, the number, length, and mass of the decoy peptides are identical to the target peptides.  Default = <code>DECOY_ 1</code>.</li>
  <li class="nobullet"><code>--max_peptide_mass &lt;float&gt;</code> &ndash; Maximum peptide mass allowed when parsing the protein sequence database. Peptides exceeding this mass will be ignored in the analysis. Default = <code>8000</code>.</li>
  <li class="nobullet"><code>--min_peptide_mass &lt;float&gt;</code> &ndash; Minimum peptide mass allowed when parsing the protein sequence database. Peptides with a lower mass will be ignored in the analysis. Default = <code>500</code>.</li>
  <li class="nobullet"><code>--truncate_prot_names &lt;integer&gt;</code> &ndash; Exports only the specified number of characters for each protein name in the Kojak output. Otherwise, if set to 0, all characters in the protein name are exported. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>CPU threads</h3>
<ul>
  <li class="nobullet"><code>--threads &lt;integer&gt;</code> &ndash; Number of threads to use when searching spectra. A value of 0 will automatically match the number of threads to the number of processing cores on the computer. Additionally, negative numbers can be used to specify threads equal to all but that number of cores. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Masses</h3>
<ul>
  <li class="nobullet"><code>--ppm_tolerance_pre &lt;float&gt;</code> &ndash; Tolerance used when determining which peptides to search for a given MS/MS spectrum based on its precursor ion mass. The unit is parts-per-million (PPM). Default = <code>10</code>.</li>
  <li class="nobullet"><code>--auto_ppm_tolerance_pre false|warn|fail</code> &ndash; Automatically estimate optimal value for the <code>ppm_tolerance_pre</code> parameter from the spectra themselves. false=no estimation, warn=try to estimate but use the default value in case of failure, fail=try to estimate and quit in case of failure. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--kojak_isotope_error &lt;integer&gt;</code> &ndash; Allows the searching of neighboring isotope peak masses for poorly resolve precursors. Up to three alternative isotope peak masses will be searched in addition to the presumed precursor peak mass to correct for errors in monoisotopic precursor peak identification. Default = <code>1</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Fragment ions</h3>
<ul>
  <li class="nobullet"><code>--fragment_bin_offset &lt;float&gt;</code> &ndash; Offset position to start the binning (0.0 to 1.0). Default = <code>0.4</code>.</li>
  <li class="nobullet"><code>--fragment_bin_size &lt;float&gt;</code> &ndash; Determines the accuracy of the scoring algorithm with smaller bins being more strict in determining matches between theoretical and observed spectral peaks. Low-resolution spectra require larger bin sizes to accommodate errors in mass accuracy of the observed peaks. Smaller bins also require more system memory, so caution must be exercised when setting this value for high-resolution spectra. For ion trap (low-res) MS/MS spectra, the recommended values is 1.0005. For high-res MS/MS, the recommended value is 0.03. Default = <code>0.03</code>.</li>
  <li class="nobullet"><code>--auto_fragment_bin_size false|warn|fail</code> &ndash; Automatically estimate optimal value for the <code>fragment_bin_size</code> parameter from the spectra themselves. false=no estimation, warn=try to estimate but use the default value in case of failure, fail=try to estimate and quit in case of failure. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--ion_series_A T|F</code> &ndash; Should A-series fragment ions be considered? Default = <code>false</code>.</li>
  <li class="nobullet"><code>--ion_series_B T|F</code> &ndash; Should B-series fragment ions be considered? Default = <code>true</code>.</li>
  <li class="nobullet"><code>--ion_series_C T|F</code> &ndash; Should C-series fragment ions be considered? Default = <code>false</code>.</li>
  <li class="nobullet"><code>--ion_series_X T|F</code> &ndash; Should X-series fragment ions be considered? Default = <code>false</code>.</li>
  <li class="nobullet"><code>--ion_series_Y T|F</code> &ndash; Should Y-series fragment ions be considered? Default = <code>true</code>.</li>
  <li class="nobullet"><code>--ion_series_Z T|F</code> &ndash; Should Z-series fragment ions be considered Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Spectral processing</h3>
<ul>
  <li class="nobullet"><code>--enrichment &lt;float&gt;</code> &ndash; Values between 0 and 1 to describe 18O atom percent excess (APE). For example, 0.25 equals 25 APE. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--MS1_centroid T|F</code> &ndash; Are the precursor ion (MS1) scans centroided? Default = <code>true</code>.</li>
  <li class="nobullet"><code>--MS2_centroid T|F</code> &ndash; Are the fragment ion (MS2) scans centroided? Default = <code>true</code>.</li>
  <li class="nobullet"><code>--MS1_resolution &lt;float&gt;</code> &ndash; Resolution at 400 m/z, value ignored if data are already centroided. Default = <code>30000</code>.</li>
  <li class="nobullet"><code>--MS2_resolution &lt;float&gt;</code> &ndash; Resolution at 400 m/z, value ignored if data are already centroided. Default = <code>25000</code>.</li>
  <li class="nobullet"><code>--min_spectrum_peaks &lt;integer&gt;</code> &ndash; Minimum number of MS/MS peaks required to proceed with analysis of a spectrum. If spectrum_processing is enabled, the peak count occurs after the spectrum is processed Default = <code>12</code>.</li>
  <li class="nobullet"><code>--max_spectrum_peaks &lt;integer&gt;</code> &ndash; Maximum number of MS/MS peaks to analyze if using the <code>spectrum_processing</code> parameter. Peaks are kept in order of intensity, starting with the most intense. Setting a value of 0 keeps all peaks. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--precursor_refinement T|F</code> &ndash; Some data files may filter out precursor scans to save space prior to searching. To analyze these files, the precursor analysis algorithms in Kojak must be disabled. It is also possible, though not always recommended, to disable these algorithms even when precursor scans are included in the data files. This parameter toggles the precursor analysis algorithms. Default = <code>true</code>.</li>
  <li class="nobullet"><code>--prefer_precursor_pred &lt;integer&gt;</code> &ndash; For some data (such as Thermo Orbitrap data), the MS/MS spectra may have a precursor mass prediction already. With this parameter, the Kojak algorithm can be set to either ignore, use, or supplement the predicted precursor information. There are three options for the parameter: 0 = Ignore all precursor mass predictions and have Kojak make new predictions using its precursor processing algorithms. 1 = Use the existing precursor mass predictions and skip further processing with Kojak. 2 = Use the existing precursor mass predictions and supplement these values with additional results of the Kojak precursor processing algorithms. The recommended value is 2. Supplementing the precursor values performs the following functions. First, the monoisotopic precursor mass may be refined to one determined from a point near the apex of the extracted ion chromatogram, with potentially better mass accuracy. Second, in cases where the monoisotopic peak mass might not have been predicted correctly in the original analysis, a second monoisotopic mass is appended to the spectrum, allowing database searching to proceed checking both possibilities. Third, in cases where there is obvious chimeric signal overlap, a spectrum will be supplemented with the potential monoisotopic peak masses of all presumed precursor ions. Regardless of this parameter setting, MS/MS spectra that do not have an existing precursor mass prediction will be analyzed to identify the monoisotopic precursor mass using functions built into Kojak. Default = <code>2</code>.</li>
  <li class="nobullet"><code>--spectrum_processing T|F</code> &ndash; The MS/MS spectrum processing function will collapse isotope distributions to the monoisotopic peak and reduce the number of peaks to analyze to the number specified with the <code>max_spectrum_peaks</code> parameter. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--kojak_instrument &lt;integer&gt;</code> &ndash; Values are: 0=Orbitrap, 1=FTICR (such as Thermo LTQ-FT). Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>param-medic options</h3>
<ul>
  <li class="nobullet"><code>--pm-min-precursor-mz &lt;float&gt;</code> &ndash; Minimum precursor m/z value to use in measurement error estimation. Default = <code>400</code>.</li>
  <li class="nobullet"><code>--pm-max-precursor-mz &lt;float&gt;</code> &ndash; Minimum precursor m/z value to use in measurement error estimation. Default = <code>1800</code>.</li>
  <li class="nobullet"><code>--pm-min-frag-mz &lt;float&gt;</code> &ndash; Minimum fragment m/z value to use in measurement error estimation. Default = <code>150</code>.</li>
  <li class="nobullet"><code>--pm-max-frag-mz &lt;float&gt;</code> &ndash; Maximum fragment m/z value to use in measurement error estimation. Default = <code>1800</code>.</li>
  <li class="nobullet"><code>--pm-min-scan-frag-peaks &lt;integer&gt;</code> &ndash; Minimum fragment peaks an MS/MS scan must contain to be used in measurement error estimation. Default = <code>40</code>.</li>
  <li class="nobullet"><code>--pm-max-precursor-delta-ppm &lt;float&gt;</code> &ndash; Maximum ppm distance between precursor m/z values to consider two scans potentially generated by the same peptide for measurement error estimation. Default = <code>50</code>.</li>
  <li class="nobullet"><code>--pm-charges &lt;string&gt;</code> &ndash; Precursor charge states to consider MS/MS spectra from, in measurement error estimation, provided as comma-separated values. Default = <code>0,2,3,4</code>.</li>
  <li class="nobullet"><code>--pm-top-n-frag-peaks &lt;integer&gt;</code> &ndash; Number of most-intense fragment peaks to consider for measurement error estimation, per MS/MS spectrum. Default = <code>30</code>.</li>
  <li class="nobullet"><code>--pm-pair-top-n-frag-peaks &lt;integer&gt;</code> &ndash; Number of fragment peaks per spectrum pair to be used in fragment error estimation. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--pm-min-common-frag-peaks &lt;integer&gt;</code> &ndash; Number of the most-intense peaks that two spectra must share in order to potentially be generated by the same peptide, for measurement error estimation. Default = <code>20</code>.</li>
  <li class="nobullet"><code>--pm-max-scan-separation &lt;integer&gt;</code> &ndash; Maximum number of scans two spectra can be separated by in order to be considered potentially generated by the same peptide, for measurement error estimation. Default = <code>1000</code>.</li>
  <li class="nobullet"><code>--pm-min-peak-pairs &lt;integer&gt;</code> &ndash; Minimum number of peak pairs (for precursor or fragment) that must be successfully paired in order to attempt to estimate measurement error distribution. Default = <code>200</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Input and output</h3>
<ul>
  <li class="nobullet"><code>--export_percolator T|F</code> &ndash; Exports results in Percolator text format (PIN format). Default = <code>true</code>.</li>
  <li class="nobullet"><code>--export_mzID T|F</code> &ndash; Exports results in mzID format. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--export_pepXML T|F</code> &ndash; Exports results in pepXML format. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--output-dir &lt;string&gt;</code> &ndash; The name of the directory where output files will be created. Default = <code>crux-output</code>.</li>
  <li class="nobullet"><code>--overwrite T|F</code> &ndash; Replace existing files if true or fail when trying to overwrite a file if false. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--parameter-file &lt;string&gt;</code> &ndash; A file containing parameters.  See the <a href="../file-formats/parameter-file.html">parameter documentation</a> page for details. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--fileroot &lt;string&gt;</code> &ndash; The fileroot string will be added as a prefix to all output file names. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--verbosity &lt;integer&gt;</code> &ndash; Specify the verbosity of the current processes. Each level prints the following messages, including all those at lower verbosity levels: 0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on the progress of execution, 40-more progress information, 50-debug info, 60-detailed debug info. Default = <code>30</code>.</li>
</ul>
</li>

</ul>
    <!-- END CONTENT -->
    </div>
  </div>
<footer class="site-footer">
  <div id="centerfooter">
    <div class="footerimportantlinks">
      <img src="../images/linkicon.png" style="width:16px; height:16px"><h3>Important links</h3>
      <ul>
        <li><a href="../faq.html">Crux FAQ</a></li>
        <li><a href="../glossary.html">Glossary of terminology</a></li>
        <li><a href="http://scholar.google.com/citations?hl=en&user=Rw9S1HIAAAAJ">Google Scholar profile</a></li>
        <li><a href="https://github.com/crux-toolkit/crux-toolkit">GitHub</a></li>
        <li><a href="../release-notes.html">Release Notes</a></li>
        <li><a href="https://mailman1.u.washington.edu/mailman/listinfo/crux-users" title="Receive announcements of new versions">Join the mailing list</a></li>
        <li><a href="http://www.apache.org/licenses/LICENSE-2.0">Apache license</a></li>
        <li><a href="http://groups.google.com/group/crux-users">Support Board</a></li>
      </ul>
    </div>
    <div class="footerimportantlinks tutoriallinks">
      <img src="../images/tutorialicon.png" style="height:16px"><h3>Tutorials</h3>
      <ul>
        <li><a href="../tutorials/install.html">Installation</a></li>
        <li><a href="../tutorials/gettingstarted.html">Getting started with Crux</a></li>
        <li><a href="../tutorials/search.html">Running a simple search using Tide and Percolator</a></li>
        <li><a href="../tutorials/customizedsearch.html">Customization and search options</a></li>
        <li><a href="../tutorials/spectralcounts.html">Using spectral-counts</a></li>
      </ul>
    </div>
    <div id="footertext">
      <p>
        The original version of Crux was written by Chris Park and Aaron Klammer
        under the supervision
        of <a href="http://www.gs.washington.edu/faculty/maccoss.htm">Prof. Michael
        MacCoss</a>
        and <a href="http://noble.gs.washington.edu/~noble">Prof. William
        Stafford Noble</a> in the Department of Genome Sciences at the
        University of Washington, Seattle.  Website by <a href="http://www.yuvalboss.com/">Yuval Boss</a>. 
        <br />The complete list of contributors
        can be found <a href="../contributors.html">here</a>.
        <br />
        <br />
        Maintenance and development of Crux is funded by the <a href="https://www.nih.gov/">National Institutes of Health</a> award P41 GM103533. 
      </p>
    </div>
  </div>
</footer>
</body>
</html>