<!DOCTYPE html>
<html>
<head>
<title>percolator</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link rel="stylesheet" type="text/css" href="../styles.css">
<script>
  window.MathJax = {
    chtml: { displayAlign: 'left' },
    options: {
      ignoreHtmlClass: 'tex2jax_ignore',
      processHtmlClass: 'tex2jax_process'
    },
    tex: {
      autoload: {
        color: [],
        colorv2: ['color']
      },
      packages: {'[+]': ['noerrors']}
    },
    loader: {load: ['[tex]/noerrors']}
  };
</script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js" id="MathJax-script"></script>
<script type="text/javascript">
  MathJax.Hub.Config({jax: ['input/TeX','output/HTML-CSS'], displayAlign: 'left'});
</script>
<script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-26136956-1']);
  _gaq.push(['_trackPageview']);
  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.8.3/jquery.min.js"></script>
<script type="text/javascript">
  // Main Menu
  $( document ).ready(function() {
      var pull 		= $('.btn');
        menu 		= $('nav ul');
        menuHeight	= menu.height();
      $(pull).on('click', function(e) {
        e.preventDefault();
        menu.slideToggle();
      });
      $(window).resize(function(){
          var w = $(window).width();
          if(w > 320 && menu.is(':hidden')) {
            menu.removeAttr('style');
          } 
      });
  });
</script>
</head>
<body>
  <div class="page-wrap">
    <nav>
      <div class="btn">
        </div>
        <img src="../images/crux-logo.png" id="logo"></a>
      <ul id="navitems">
          <li><a href="../index.html">Home</a></li>
          <li><a href="../download.html">Download</a></li>
          <li><a href="../fileformats.html">File Formats</a></li>
            <li><a href="http://groups.google.com/group/crux-users">Contact</a></li> <!--Link to google support board-->
        </ul>
    </nav>
    <div id="content" class="autogenerated">
    <!-- START CONTENT -->
<h1>percolator</h1>
<h2>Usage:</h2>
<p><code>crux percolator [options] &lt;peptide-spectrum matches&gt;</code></p>
<h2>Description:</h2>
<p>Percolator is a semi-supervised learning algorithm that dynamically learns to separate target from decoy peptide-spectrum matches (PSMs). The algorithm is described in this article:</p><blockquote> Lukas K&auml;ll, Jesse Canterbury, Jason Weston, William Stafford Noble and Michael J. MacCoss. <a href="http://noble.gs.washington.edu/papers/kall2007semi-supervised.html">&quot;Semi-supervised learning for peptide identification from shotgun proteomics datasets.&quot;</a> <em>Nature Methods</em>. 4(11):923-925, 2007.</blockquote><p>Percolator requires as input two collections of PSMs, one set derived from matching observed spectra against real (&quot;target&quot;) peptides, and a second derived from matching the same spectra against &quot;decoy&quot; peptides. The output consists of ranked lists of PSMs, peptides and proteins. Peptides and proteins are assigned two types of statistical confidence estimates: q-values and posterior error probabilities.</p><p>The features used by Percolator to represent each PSM are summarized <a href="../file-formats/features.html">here</a>.</p><p>Percolator also includes code from <a href="http://noble.gs.washington.edu/proj/fido/">Fido</a>, whch performs protein-level inference. The Fido algorithm is described in this article:</p><blockquote>Oliver Serang, Michael J. MacCoss and William Stafford Noble. <a href="http://pubs.acs.org/doi/abs/10.1021/pr100594k">&quot;Efficient marginalization to compute protein posterior probabilities from shotgun mass spectrometry data.&quot;</a> <em>Journal of Proteome Research</em>. 9(10):5346-5357, 2010.</blockquote><p>Crux includes code from <a href="http://percolator.ms">Percolator</a>. Crux Percolator differs from the stand-alone version of Percolator in the following respects:</p><ul><li>In addition to the native Percolator XML file format, Crux Percolator supports additional input file formats (SQT, PepXML, tab-delimited text).To maintain consistency with the rest of the Crux commands, Crux Percolator uses different parameter syntax than the stand-alone version of Percolator.</li><li>Like the rest of the Crux commands, Crux Percolator writes its files to an output directory, logs all standard error messages to a log file, and is capable of reading parameters from a parameter file.</li><li>Reading from XML and stdin are not supported at this time.</li></ul>
<h2>Input:</h2>
<ul>
  <li><code>peptide-spectrum matches</code> &ndash; One or more collections of target and decoy peptide-spectrum matches (PSMs). Input may be in one of four formats: PIN, SQT, pepXML, or <a href="../file-formats/txt-format.html">Crux tab-delimited text</a>. Note that if the input is provided as SQT, pepXML, or Crux tab-delimited text, then a PIN file will be generated in the output directory prior to execution. Crux determines the format of the input file by examining its filename extension.<br>For PIN files, target and decoy PSMs are assumed to appear in the same file.  For other file types, decoy PSMs can be provided to Percolator in two ways: either as a separate file or embedded within the same file as the target PSMs. Percolator will first search for target PSMs in a separate file. The decoy file name is constructed from the target name by replacing "target" with "decoy". For example, if search.target.txt is provided as input, then Percolator will search for a corresponding file named search.decoy.txt. If no decoy file is found, then Percolator will assume that the given input file contains a mix of target and decoy PSMs. Within this file, decoys are identified using a prefix (specified via --decoy-prefix) on the protein name.</li>
</ul>
<h2>Output:</h2>
<p>The program writes files to the folder <code>crux-output</code> by default. The name of the output folder can be set by the user using the <code>--output-dir</code> option. The following files will be created:
<ul>
  <li><code>percolator.target.proteins.txt</code> &ndash; a tab-delimited file containing the target protein matches. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.decoy.proteins.txt</code> &ndash; a tab-delimited file containing the decoy protein matches. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.target.peptides.txt</code> &ndash; a tab-delimited file containing the target peptide matches. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.decoy.peptides.txt</code> &ndash; a tab-delimited file containing the decoy peptide matches. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.target.psms.txt</code> &ndash; a tab-delimited file containing the target PSMs. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.decoy.psms.txt</code> &ndash; a tab-delimited file containing the decoy PSMs. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.params.txt</code> &ndash; a file containing the name and value of all parameters for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
  <li><code>percolator.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to standard error.</li>
</ul>
<h2>Options:</h2>
<ul style="list-style-type: none;">
<li class="nobullet">
<h3>percolator options</h3>
<ul>
  <li class="nobullet"><code>--max-charge-feature &lt;integer&gt;</code> &ndash; Specifies the maximum charge state feature.  When set to zero, use the maximum observed charge state. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--no-terminate T|F</code> &ndash; Do not stop execution when encountering questionable SVM inputs or results. "percolator.weights.txt". Default = <code>false</code>.</li>
  <li class="nobullet"><code>--protein-name-separator &lt;string&gt;</code> &ndash; Determines the character to separate the protein IDs in the tab-delimited output format  Default = <code>,</code>.</li>
  <li class="nobullet"><code>--spectral-counting-fdr &lt;float&gt;</code> &ndash; Report the number of unique PSMs and total (including shared peptides) PSMs as two extra columns in the protein tab-delimited output. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--train-best-positive T|F</code> &ndash; Enforce that, for each spectrum, at most one PSM is included in the positive set during each training iteration. Note that if the user only provides one PSM per spectrum, then this option will have no effect. Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Protein inference options</h3>
<ul>
  <li class="nobullet"><code>--picked-protein &lt;string&gt;</code> &ndash; Use the picked protein-level FDR to infer protein probabilities, provide the fasta file as the argument to this flag. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--protein-enzyme no_enzyme|elastase|pepsin|proteinasek|thermolysin|trypsinp|chymotrypsin|lys-n|lys-c|arg-c|asp-n|glu-c|lysarginase|trypsin</code> &ndash; Type of enzyme Default = <code>trypsin</code>.</li>
  <li class="nobullet"><code>--protein-report-duplicates T|F</code> &ndash; If multiple database proteins contain exactly the same set of peptides, then Percolator will randomly discard all but one of the proteins. If this option is set, then the IDs of these duplicated proteins will be reported as a comma-separated list. Not available for Fido. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--protein-report-fragments T|F</code> &ndash; By default, if the peptides associated with protein A are a proper subset of the peptides associated with protein B, then protein A is eliminated and all the peptides are considered as evidence for protein B. Note that this filtering is done based on the complete set of peptides in the database, not based on the identified peptides in the search results. Alternatively, if this option is set and if all of the identified peptides associated with protein B are also associated with protein A, then Percolator will report a comma-separated list of protein IDs, where the full-length protein B is first in the list and the fragment protein A is listed second. Not available for Fido. Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Fido options</h3>
<ul>
  <li class="nobullet"><code>--fido-alpha &lt;float&gt;</code> &ndash; Specify the probability with which a present protein emits an associated peptide. Set by grid search (see --fido-gridsearch-depth parameter) if not specified. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fido-beta &lt;float&gt;</code> &ndash; Specify the probability of the creation of a peptide from noise. Set by grid search (see --fido-gridsearch-depth parameter) if not specified. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fido-empirical-protein-q T|F</code> &ndash; Estimate empirical p-values and q-values for proteins using target-decoy analysis. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--fido-fast-gridsearch &lt;float&gt;</code> &ndash; Apply the specified threshold to PSM, peptide and protein probabilities to obtain a faster estimate of the alpha, beta and gamma parameters. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fido-gamma &lt;float&gt;</code> &ndash; Specify the prior probability that a protein is present in the sample. Set by grid search (see --fido-gridsearch-depth parameter) if not specified. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fido-gridsearch-depth &lt;integer&gt;</code> &ndash; Set depth of the grid search for alpha, beta and gamma estimation. The values considered, for each possible value of the --fido-gridsearch-depth parameter, are as follows:<ul><li>0: alpha = {0.01, 0.04, 0.09, 0.16, 0.25, 0.36, 0.5}; beta = {0.0, 0.01, 0.15, 0.025, 0.035, 0.05, 0.1}; gamma = {0.1, 0.25, 0.5, 0.75}.</li><li>1: alpha = {0.01, 0.04, 0.09, 0.16, 0.25, 0.36}; beta = {0.0, 0.01, 0.15, 0.025, 0.035, 0.05}; gamma = {0.1, 0.25, 0.5}.</li><li>2: alpha = {0.01, 0.04, 0.16, 0.25, 0.36}; beta = {0.0, 0.01, 0.15, 0.030, 0.05}; gamma = {0.1, 0.5}.</li><li>3: alpha = {0.01, 0.04, 0.16, 0.25, 0.36}; beta = {0.0, 0.01, 0.15, 0.030, 0.05}; gamma = {0.5}.</li></ul> Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fido-gridsearch-mse-threshold &lt;float&gt;</code> &ndash; Q-value threshold that will be used in the computation of the MSE and ROC AUC score in the grid search. Default = <code>0.05</code>.</li>
  <li class="nobullet"><code>--fido-no-split-large-components T|F</code> &ndash; Do not approximate the posterior distribution by allowing large graph components to be split into subgraphs. The splitting is done by duplicating peptides with low probabilities. Splitting continues until the number of possible configurations of each subgraph is below 2^18 Default = <code>false</code>.</li>
  <li class="nobullet"><code>--fido-protein-truncation-threshold &lt;float&gt;</code> &ndash; To speed up inference, proteins for which none of the associated peptides has a probability exceeding the specified threshold will be assigned probability = 0. Default = <code>0.01</code>.</li>
  <li class="nobullet"><code>--protein T|F</code> &ndash; Use the Fido algorithm to infer protein probabilities. Must be true to use any of the Fido options. Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>General options</h3>
<ul>
  <li class="nobullet"><code>--only-psms T|F</code> &ndash; Report results only at the PSM level.  This flag causes Percolator to skip the step that selects the top-scoring PSM per peptide; hence, peptide-level results are left out and only PSM-level results are reported. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--search-input auto|separate|concatenated</code> &ndash; Specify the type of target-decoy search. Using 'auto', percolator attempts to detect the search type automatically.  Using 'separate' specifies two searches: one against target and one against decoy protein db. Using 'concatenated' specifies a single search on concatenated target-decoy protein db. Default = <code>auto</code>.</li>
  <li class="nobullet"><code>--tdc T|F</code> &ndash; Use target-decoy competition to assign q-values and PEPs. When set to F, the mix-max method, which estimates the proportion pi0 of incorrect target PSMs, is used instead. Default = <code>true</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>SVM training options</h3>
<ul>
  <li class="nobullet"><code>--c-neg &lt;float&gt;</code> &ndash; Penalty for mistake made on negative examples. If not specified, then this value is set by cross validation over {0.1, 1, 10}. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--c-pos &lt;float&gt;</code> &ndash; Penalty for mistakes made on positive examples. If this value is set to 0, then it is set via cross validation over the values {0.1, 1, 10}, selecting the value that yields the largest number of PSMs identified at the q-value threshold set via the --test-fdr parameter. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--maxiter &lt;integer&gt;</code> &ndash; Maximum number of iterations for training. Default = <code>10</code>.</li>
  <li class="nobullet"><code>--percolator-seed &lt;string&gt;</code> &ndash; When given a unsigned integer value seeds the random number generator with that value. When given the string "time" seeds the random number generator with the system time. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--quick-validation T|F</code> &ndash; Quicker execution by reduced internal cross-validation. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--static T|F</code> &ndash; Use the provided initial weights as a static model. If used, the --init-weights option must be specified. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--subset-max-train &lt;integer&gt;</code> &ndash; Only train Percolator on a subset of PSMs, and use the resulting score vector to evaluate the other PSMs. Recommended when analyzing huge numbers (>1 million) of PSMs. When set to 0, all PSMs are used for training as normal. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--test-each-iteration T|F</code> &ndash; Measure performance on test set each iteration. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--test-fdr &lt;float&gt;</code> &ndash; False discovery rate threshold used in selecting hyperparameters during internal cross-validation and for reporting the final results. Default = <code>0.01</code>.</li>
  <li class="nobullet"><code>--train-fdr &lt;float&gt;</code> &ndash; False discovery rate threshold to define positive examples in training. Default = <code>0.01</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>SVM feature input options</h3>
<ul>
  <li class="nobullet"><code>--default-direction &lt;string&gt;</code> &ndash; In its initial round of training, Percolator uses one feature to induce a ranking of PSMs. By default, Percolator will select the feature that produces the largest set of target PSMs at a specified FDR threshold (cf. --train-fdr). This option allows the user to specify which feature is used for the initial ranking, using the name as a string from <a href="../file-formats/features.html">this table</a>. The name can be preceded by a hyphen (e.g. "-XCorr") to indicate that a lower value is better. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--init-weights &lt;string&gt;</code> &ndash; Read the unnormalized initial weights from the third line of the given file. This can be the output of the --output-weights option from a previous Percolator analysis. Note that the weights must be in the same order as features in the PSM input file(s) Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--klammer T|F</code> &ndash; Use retention time features calculated as in "Improving tandem mass spectrum identification using peptide retention time prediction across diverse chromatography conditions" by Klammer AA, Yi X, MacCoss MJ and Noble WS. (<em>Analytical Chemistry</em>. 2007 Aug 15;79(16):6111-8.). Default = <code>false</code>.</li>
  <li class="nobullet"><code>--output-weights T|F</code> &ndash; Output final weights to a file named "percolator.weights.txt". Default = <code>false</code>.</li>
  <li class="nobullet"><code>--override T|F</code> &ndash; By default, Percolator will examine the learned weights for each feature, and if the weight appears to be problematic, then percolator will discard the learned weights and instead employ a previously trained, static score vector. This switch allows this error checking to be overriden. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--unitnorm T|F</code> &ndash; Use unit normalization (i.e., linearly rescale each PSM's feature vector to have a Euclidean length of 1), instead of standard deviation normalization. Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Input and output</h3>
<ul>
  <li class="nobullet"><code>--decoy-prefix &lt;string&gt;</code> &ndash; Specifies the prefix of the protein names that indicate a decoy. Default = <code>decoy_</code>.</li>
  <li class="nobullet"><code>--decoy-xml-output T|F</code> &ndash; Include decoys (PSMs, peptides, and/or proteins) in the XML output. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--feature-file-out T|F</code> &ndash; Output the computed features in <a href="../file-formats/features.html">tab-delimited Percolator input (.pin) format</a>. The features will be normalized, using either unit norm or standard deviation normalization (depending upon the value of the unit-norm option). Default = <code>false</code>.</li>
  <li class="nobullet"><code>--fileroot &lt;string&gt;</code> &ndash; The fileroot string will be added as a prefix to all output file names. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--mzid-output T|F</code> &ndash; Output an mzIdentML results file to the output directory. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--output-dir &lt;string&gt;</code> &ndash; The name of the directory where output files will be created. Default = <code>crux-output</code>.</li>
  <li class="nobullet"><code>--overwrite T|F</code> &ndash; Replace existing files if true or fail when trying to overwrite a file if false. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--parameter-file &lt;string&gt;</code> &ndash; A file containing parameters.  See the <a href="../file-formats/parameter-file.html">parameter documentation</a> page for details. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--pout-output T|F</code> &ndash; Output a Percolator <a href="https://github.com/percolator/percolator/blob/master/src/xml/percolator_out.xsd">pout.xml</a> format results file to the output directory. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--top-match &lt;integer&gt;</code> &ndash; Specify the number of matches to report for each spectrum. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--txt-output T|F</code> &ndash; Output a tab-delimited results file to the output directory. Default = <code>true</code>.</li>
  <li class="nobullet"><code>--verbosity &lt;integer&gt;</code> &ndash; Specify the verbosity of the current processes. Each level prints the following messages, including all those at lower verbosity levels: 0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on the progress of execution, 40-more progress information, 50-debug info, 60-detailed debug info. Default = <code>30</code>.</li>
</ul>
</li>

</ul>
    <!-- END CONTENT -->
    </div>
  </div>
<footer class="site-footer">
  <div id="centerfooter">
    <div class="footerimportantlinks">
      <img src="../images/linkicon.png" style="width:16px; height:16px"><h3>Important links</h3>
      <ul>
        <li><a href="../faq.html">Crux FAQ</a></li>
        <li><a href="../glossary.html">Glossary of terminology</a></li>
        <li><a href="http://scholar.google.com/citations?hl=en&user=Rw9S1HIAAAAJ">Google Scholar profile</a></li>
        <li><a href="https://github.com/crux-toolkit/crux-toolkit">GitHub</a></li>
        <li><a href="../release-notes.html">Release Notes</a></li>
        <li><a href="https://mailman1.u.washington.edu/mailman/listinfo/crux-users" title="Receive announcements of new versions">Join the mailing list</a></li>
        <li><a href="http://www.apache.org/licenses/LICENSE-2.0">Apache license</a></li>
        <li><a href="http://groups.google.com/group/crux-users">Support Board</a></li>
      </ul>
    </div>
    <div class="footerimportantlinks tutoriallinks">
      <img src="../images/tutorialicon.png" style="height:16px"><h3>Tutorials</h3>
      <ul>
        <li><a href="../tutorials/install.html">Installation</a></li>
        <li><a href="../tutorials/gettingstarted.html">Getting started with Crux</a></li>
        <li><a href="../tutorials/search.html">Running a simple search using Tide and Percolator</a></li>
        <li><a href="../tutorials/customizedsearch.html">Customization and search options</a></li>
        <li><a href="../tutorials/spectralcounts.html">Using spectral-counts</a></li>
      </ul>
    </div>
    <div id="footertext">
      <p>
        The original version of Crux was written by Chris Park and Aaron Klammer
        under the supervision
        of <a href="http://www.gs.washington.edu/faculty/maccoss.htm">Prof. Michael
        MacCoss</a>
        and <a href="http://noble.gs.washington.edu/~noble">Prof. William
        Stafford Noble</a> in the Department of Genome Sciences at the
        University of Washington, Seattle.  Website by <a href="http://www.yuvalboss.com/">Yuval Boss</a>. 
        <br />The complete list of contributors
        can be found <a href="../contributors.html">here</a>.
        <br />
        <br />
        Maintenance and development of Crux is funded by the <a href="https://www.nih.gov/">National Institutes of Health</a> award P41 GM103533. 
      </p>
    </div>
  </div>
</footer>
</body>
</html>