<!DOCTYPE html>
<html>
<head>
<title>pipeline</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link rel="stylesheet" type="text/css" href="../styles.css">
<script type="text/javascript"
  src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/javascript">
  MathJax.Hub.Config({jax: ['input/TeX','output/HTML-CSS'], displayAlign: 'left'});
</script>
<script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-26136956-1']);
  _gaq.push(['_trackPageview']);
  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>
<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.8.3/jquery.min.js"></script>
<script type="text/javascript">
  // Main Menu
  $( document ).ready(function() {
      var pull 		= $('.btn');
        menu 		= $('nav ul');
        menuHeight	= menu.height();
      $(pull).on('click', function(e) {
        e.preventDefault();
        menu.slideToggle();
      });
      $(window).resize(function(){
          var w = $(window).width();
          if(w > 320 && menu.is(':hidden')) {
            menu.removeAttr('style');
          } 
      });
  });
</script>
</head>
<body>
  <div class="page-wrap">
    <nav>
      <div class="btn">
        </div>
        <img src="../images/crux-logo.png" id="logo"></a>
      <ul id="navitems">
          <li><a href="../index.html">Home</a></li>
          <li><a href="../download.html">Download</a></li>
          <li><a href="../fileformats.html">File Formats</a></li>
            <li><a href="http://groups.google.com/group/crux-users">Contact</a></li> <!--Link to google support board-->
        </ul>
    </nav>
    <div id="content" class="autogenerated">
    <!-- START CONTENT -->
<h1>pipeline</h1>
<h2>Usage:</h2>
<p><code>crux pipeline [options] &lt;mass spectra&gt; &lt;peptide source&gt;</code></p>
<h2>Description:</h2>
<p>Given one or more sets of tandem mass spectra as well as a protein database, this command runs a series of Crux tools and reports all of the results in a single output directory. There are three steps in the pipeline:</p><ol><li><a href="bullseye.html">Bullseye</a> to assign high-resolution precursor m/z values to MS/MS data. This step is optional.</li><li>Database searching using either <a href="tide-search.html">Tide-search</a> or <a href="comet.html">Comet</a>. The database can be provided as a file in FASTA format, or additionally, an index as produced by <a href="tide-index.html">tide-index</a>.</li><li>Post-processing using either <a href="assign-confidence.html">assign-confidence</a> or <a href="percolator.html">Percolator</a>.</li></ol><p>All of the command line options associated with the individual tools in the pipeline can be used with the <code>pipeline</code> command.</p>
<h2>Input:</h2>
<ul>
  <li><code>mass spectra</code> &ndash; The name of the file(s) from which to parse the fragmentation spectra, in any of the <a href="http://proteowizard.sourceforge.net/tools.shtml">file formats supported by ProteoWizard</a>. Alteratively, with Tide-search, these files may be binary spectrum files produced by a previous run of <code>crux tide-search</code> using the <code>store-spectra</code> parameter. Multiple files can be included on the command line (space delimited), prior to the name of the database.</li>
  <li><code>peptide source</code> &ndash; Either the name of a file in fasta format from which to retrieve proteins and peptides or an index created by a previous run of <code>crux tide-index</code> (for Tide searching).</li>
</ul>
<h2>Output:</h2>
<p>The program writes files to the folder <code>crux-output</code> by default. The name of the output folder can be set by the user using the <code>--output-dir</code> option. The following files will be created:
<ul>
  <li><code>bullseye.pid.<format></code> &ndash; a file containing the fragmentation spectra for which accurate masses were successfully inferred. Unless otherwise specified (with the --spectrum-format option), the output file format is ".ms2". Note that if the output format is ".ms2," then a single spectrum may have multiple "Z" lines, each indicating a charge state and accurate mass. In addition, Bullseye inserts an "I" line (for charge-dependent analysis) corresponding to each "Z" line. The "I" line contains "EZ" in the second column, the charge and mass from the associated "Z" line in the third and fourth colummns, followed by the chromatographic apex and the intensity at the chromatographic apex.</li>
  <li><code>bullseye.no-pid.<format></code> &ndash; a file containing the fragmentation spectra for which accurate masses were not inferred.</li>
  <li><code>hardklor.mono.txt</code> &ndash; a tab-delimited text file containing one line for each isotope distribution, as described <a href="hardklor.html">here</a>.</li>
  <li><code>bullseye.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
  <li><code>bullseye.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to standard error.</li>
  <li><code>tide-search.target.txt</code> &ndash; a tab-delimited text file containing the target PSMs. See <a href="../file-formats/txt-format.html">txt file format</a> for a list of the fields.</li>
  <li><code>tide-search.decoy.txt</code> &ndash; a tab-delimited text file containing the decoy PSMs. This file will only be created if the index was created with decoys.</li>
  <li><code>tide-search.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other Crux programs.</li>
  <li><code>tide-search.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to the screen during execution.</li>
  <li><code>comet.target.txt</code> &ndash; a tab-delimited text file containing the target PSMs. See <a href="../file-formats/txt-format.html">txt file format</a> for a list of the fields.</li>
  <li><code>comet.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
  <li><code>comet.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to standard error.</li>
  <li><code>percolator.target.proteins.txt</code> &ndash; a tab-delimited file containing the target protein matches. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.decoy.proteins.txt</code> &ndash; a tab-delimited file containing the decoy protein matches. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.target.peptides.txt</code> &ndash; a tab-delimited file containing the target peptide matches. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.decoy.peptides.txt</code> &ndash; a tab-delimited file containing the decoy peptide matches. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.target.psms.txt</code> &ndash; a tab-delimited file containing the target PSMs. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.decoy.psms.txt</code> &ndash; a tab-delimited file containing the decoy PSMs. See <a href="../file-formats/txt-format.html">here</a> for a list of the fields.</li>
  <li><code>percolator.params.txt</code> &ndash; a file containing the name and value of all parameters for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
  <li><code>percolator.pep.xml</code> &ndash; a file containing the PSMs in <a href="http://tools.proteomecenter.org/wiki/index.php?title=Formats:pepXML">pepXML format</a>. This file can be used as input to some of the tools in the <a href="http://tools.proteomecenter.org/wiki/index.php?title=Software:TPP">Transproteomic Pipeline</a>.</li>
  <li><code>percolator.mzid</code> &ndash; a file containing the protein, peptide, and spectrum matches in <a href="http://www.psidev.info/mzidentml">mzIdentML format</a>.</li>
  <li><code>percolator.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to standard error.</li>
  <li><code>assign-confidence.target.txt</code> &ndash; a <a href="../file-formats/txt-format.html">tab-delimited text file</a> that contains the targets, sorted by score. The file will contain one new column, named "&lt;method&gt; q-value", where &lt;method&gt; is either "tdc" or "mix-max".</li>
  <li><code>assign-confidence.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to stderr.</li>
  <li><code>assign-confidence.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
</ul>
<h2>Options:</h2>
<ul style="list-style-type: none;">
<li class="nobullet">
<h3>pipeline options</h3>
<ul>
  <li class="nobullet"><code>--bullseye T|F</code> &ndash; Run the Bullseye algorithm on the given MS data, using it to assign high-resolution precursor values to the MS/MS data. If a spectrum file ends with .ms2 or .cms2, matching .ms1/.cms1 files will be used as the MS1 file. Otherwise, it is assumed that the spectrum file contains both MS1 and MS2 scans. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--search-engine comet|tide-search</code> &ndash; Specify which search engine to use. Default = <code>tide-search</code>.</li>
  <li class="nobullet"><code>--post-processor percolator|assign-confidence|none</code> &ndash; Specify which post-processor to apply to the search results. Default = <code>percolator</code>.</li>
  <li class="nobullet"><code>--memory-limit &lt;integer&gt;</code> &ndash; The maximum amount of memory (i.e., RAM), in GB, to be used by tide-index. Default = <code>4</code>.</li>
  <li class="nobullet"><code>--auto-modifications-spectra &lt;string&gt;</code> &ndash; Specify the spectra file to be used for modification inference when the auto-modifications option is enabled. Multiple files may be separated by commas. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--use-tailor-calibration T|F</code> &ndash; Fast, but heuristic PSM score calibration as described in <a href="https://pubmed.ncbi.nlm.nih.gov/32175744/">this article</a>. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--max-charge-feature &lt;integer&gt;</code> &ndash; Specifies the maximum charge state feature.  When set to zero, use the maximum observed charge state. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--protein-name-separator &lt;string&gt;</code> &ndash; Determines the character to separate the protein IDs in the tab-delimited output format  Default = <code>,</code>.</li>
  <li class="nobullet"><code>--spectral-counting-fdr &lt;float&gt;</code> &ndash; Report the number of unique PSMs and total (including shared peptides) PSMs as two extra columns in the protein tab-delimited output. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--train-best-positive T|F</code> &ndash; Enforce that, for each spectrum, at most one PSM is included in the positive set during each training iteration. Note that if the user only provides one PSM per spectrum, then this option will have no effect. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--estimation-method mix-max|tdc|peptide-level</code> &ndash; Specify the method used to estimate q-values.  The mix-max procedure or target-decoy competition apply to PSMs. The peptide-level option eliminates any PSM for which there exists a better scoring PSM involving the same peptide, and then uses decoys to assign confidence estimates. Default = <code>tdc</code>.</li>
  <li class="nobullet"><code>--score &lt;string&gt;</code> &ndash; Specify the column (for tab-delimited input) or tag (for XML input) used as input to the q-value estimation procedure. If this parameter is unspecified, then the program searches for "xcorr score", "evalue" (comet), "exact p-value" score fields in this order in the input file.  Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--sidak T|F</code> &ndash; Adjust the score using the Sidak adjustment and reports them in a new column in the output file. Note that this adjustment only makes sense if the given scores are p-values, and that it requires the presence of the "distinct matches/spectrum" feature for each PSM. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--top-match-in &lt;integer&gt;</code> &ndash; Specify the maximum rank to allow when parsing results files. Matches with ranks higher than this value will be ignored (a value of zero allows matches with any rank). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--combine-charge-states T|F</code> &ndash; Specify this parameter to T in order to combine charge states with peptide sequencesin peptide-centric search. Works only if estimation-method = peptide-level. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--combine-modified-peptides T|F</code> &ndash; Specify this parameter to T in order to treat peptides carrying different or no modifications as being the same. Works only if estimation = peptide-level. Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Identifying PPIDs in MS1 spectra</h3>
<ul>
  <li class="nobullet"><code>--max-persist &lt;float&gt;</code> &ndash; Ignore PPIDs that persist for longer than this length of time in the MS1 spectra. The unit of time is whatever unit is used in your data file (usually minutes). These PPIDs are considered contaminants. Default = <code>2</code>.</li>
  <li class="nobullet"><code>--persist-tolerance &lt;float&gt;</code> &ndash; Set the mass tolerance (+/-ppm) for finding PPIDs in consecutive MS1 scans. Default = <code>10</code>.</li>
  <li class="nobullet"><code>--gap-tolerance &lt;integer&gt;</code> &ndash; Allowed gap size when checking for PPIDs across consecutive MS1 scans. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--scan-tolerance &lt;integer&gt;</code> &ndash; Total number of MS1 scans over which a PPID must be observed to be considered real. Gaps in persistence are allowed by setting --gap-tolerance. Default = <code>3</code>.</li>
  <li class="nobullet"><code>--bullseye-max-mass &lt;float&gt;</code> &ndash; Only consider PPIDs below this maximum mass in daltons. Default = <code>8000</code>.</li>
  <li class="nobullet"><code>--bullseye-min-mass &lt;float&gt;</code> &ndash; Only consider PPIDs above this minimum mass in daltons. Default = <code>600</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Matching PPIDs to MS2 spectra</h3>
<ul>
  <li class="nobullet"><code>--exact-match T|F</code> &ndash; When true, require an exact match (as defined by --exact-tolerance) between the center of the precursor isolation window in the MS2 scan and the base isotopic peak of the PPID. If this option is set to false and no exact match is observed, then attempt to match using a wider m/z tolerance. This wider tolerance is calculated using the PPID's monoisotopic mass and charge (the higher the charge, the smaller the window). Default = <code>false</code>.</li>
  <li class="nobullet"><code>--exact-tolerance &lt;float&gt;</code> &ndash; Set the tolerance (+/-ppm) for --exact-match. Default = <code>10</code>.</li>
  <li class="nobullet"><code>--retention-tolerance &lt;float&gt;</code> &ndash; Set the tolerance (+/-units) around the retention time over which a PPID can be matches to the MS2 spectrum. The unit of time is whatever unit is used in your data file (usually minutes). Default = <code>0.5</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Peptide properties</h3>
<ul>
  <li class="nobullet"><code>--clip-nterm-methionine T|F</code> &ndash; When set to T, for each protein that begins with methionine, tide-index will put two copies of the leading peptide into the index, with and without the N-terminal methionine. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--isotopic-mass average|mono</code> &ndash; Specify the type of isotopic masses to use when calculating the peptide mass. Default = <code>mono</code>.</li>
  <li class="nobullet"><code>--max-length &lt;integer&gt;</code> &ndash; The maximum length of peptides to consider. Default = <code>50</code>.</li>
  <li class="nobullet"><code>--max-mass &lt;float&gt;</code> &ndash; The maximum mass (in Da) of peptides to consider. Default = <code>7200</code>.</li>
  <li class="nobullet"><code>--min-length &lt;integer&gt;</code> &ndash; The minimum length of peptides to consider. Default = <code>6</code>.</li>
  <li class="nobullet"><code>--min-mass &lt;float&gt;</code> &ndash; The minimum mass (in Da) of peptides to consider. Default = <code>200</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Amino acid modifications</h3>
<ul>
  <li class="nobullet"><code>--cterm-peptide-mods-spec &lt;string&gt;</code> &ndash; Specify peptide c-terminal modifications. See nterm-peptide-mods-spec for syntax. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--cterm-protein-mods-spec &lt;string&gt;</code> &ndash; [[html:Specifies C-terminal static and variable mass modifications on proteins.Mod specification syntax is the same as for peptide mods (see nterm-peptide-mods-spec option),but these mods are applied only to peptide C-terminals that are also protein terminals.If variable modification are provided for both peptide and protein terminal, they will be applied one at a time.  Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--max-mods &lt;integer&gt;</code> &ndash; The maximum number of modifications that can be applied to a single peptide. Default = <code>255</code>.</li>
  <li class="nobullet"><code>--min-mods &lt;integer&gt;</code> &ndash; The minimum number of modifications that can be applied to a single peptide. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--mod-precision &lt;integer&gt;</code> &ndash; Set the precision for modifications as written to .txt files. Default = <code>4</code>.</li>
  <li class="nobullet"><code>--mods-spec &lt;string&gt;</code> &ndash; The general form of a modification specification has three components, as exemplified by <span style="color: red;">1</span><span style="color: green;">STY</span>+<span style="color: blue">79.966331</span>.<br>The three components are: [<span style="color: red;">max_per_peptide</span>]<span style="color: green;">residues</span>[+/-]<span style-"color: blue;">mass_change</span><br>In the example, <span style="color: red;">max_per_peptide</span> is <span style="color: red;">1</span>, <span style="color: green;">residues</span> are <span style="color: green;">STY</span>, and <span style="color: blue;">mass_change</span> is <span style="color: blue;">+79.966331</span>. To specify a static modification, the number preceding the amino acid must be omitted; i.e., <span style="color: green;">C</span>+<span style="color: blue;">57.02146</span> specifies a static modification of 57.02146 Da to cysteine. Note that Tide allows at most one modification per amino acid.  Also, the default modification (C+57.02146) will be added to every mods-spec string unless an explicit C+0 is included. Default = <code>C+57.02146</code>.</li>
  <li class="nobullet"><code>--nterm-peptide-mods-spec &lt;string&gt;</code> &ndash; Specify peptide n-terminal modifications. Like --mods-spec, this specification has three components, but with a slightly different syntax. The <span style="color: red;">max_per_peptide</span> can be either "1", in which case it defines a variable terminal modification, or missing, in which case the modification is static. The <span style="color: green;">residues</span> field indicates which amino acids are subject to the modification, with the residue <span style="color: green;">X</span> corresponding to any amino acid. Finally, <span style="color: blue;">added_mass</span> is defined as before. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--nterm-protein-mods-spec &lt;string&gt;</code> &ndash;  [[html:Same as cterm-protein-mods-spec, but for  the protein N-terminal. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--auto-modifications T|F</code> &ndash; Automatically infer modifications from the spectra themselves. Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Decoy database generation</h3>
<ul>
  <li class="nobullet"><code>--allow-dups T|F</code> &ndash; Prevent duplicate peptides between the target and decoy databases. When set to "F", the program keeps all target and previously generated decoy peptides in memory. A shuffled decoy will be re-shuffled multiple times to avoid duplication. If a non-duplicated peptide cannot be generated, the decoy is skipped entirely. When set to "T", every decoy is added to the database without checking for duplication. This option reduces the memory requirements significantly. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--decoy-format none|shuffle|peptide-reverse</code> &ndash; Include a decoy version of every peptide by shuffling or reversing the target sequence or protein. In shuffle or peptide-reverse mode, each peptide is either reversed or shuffled, leaving the N-terminal and C-terminal amino acids in place. Note that peptides appear multiple times in the target database are only shuffled once. In peptide-reverse mode, palindromic peptides are shuffled. Also, if a shuffled peptide produces an overlap with the target or decoy database, then the peptide is re-shuffled up to 5 times. Note that, despite this repeated shuffling, homopolymers will appear in both the target and decoy database. Default = <code>shuffle</code>.</li>
  <li class="nobullet"><code>--keep-terminal-aminos N|C|NC|none</code> &ndash; When creating decoy peptides using decoy-format=shuffle or decoy-format=peptide-reverse, this option specifies whether the N-terminal and C-terminal amino acids are kept in place or allowed to be shuffled or reversed. For a target peptide "EAMPK" with decoy-format=peptide-reverse, setting keep-terminal-aminos to "NC" will yield "EPMAK"; setting it to "C" will yield "PMAEK"; setting it to "N" will yield "EKPMA"; and setting it to "none" will yield "KPMAE". Default = <code>NC</code>.</li>
  <li class="nobullet"><code>--num-decoys-per-target &lt;integer&gt;</code> &ndash; The number of decoys to generate per target. When set to a value n, then with concat=F tide-search will output one target and n decoys. The resulting files can be used to run the "average target-decoy competition" method in assign-confidence. This parameter only applies when decoy-format=shuffle and should always be used in combination with allow-dups=T. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--seed &lt;string&gt;</code> &ndash; When given a unsigned integer value seeds the random number generator with that value. When given the string "time" seeds the random number generator with the system time. Default = <code>1</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Enzymatic digestion</h3>
<ul>
  <li class="nobullet"><code>--custom-enzyme &lt;string&gt;</code> &ndash; Specify rules for in silico digestion of protein sequences. Overrides the enzyme option. Two lists of residues are given enclosed in square brackets or curly braces and separated by a |. The first list contains residues required/prohibited before the cleavage site and the second list is residues after the cleavage site. If the residues are required for digestion, they are in square brackets, '[' and ']'. If the residues prevent digestion, then they are enclosed in curly braces, '{' and '}'. Use X to indicate all residues. For example, trypsin cuts after R or K but not before P which is represented as [RK]|{P}. AspN cuts after any residue but only before D which is represented as [X]|[D]. To prevent the sequences from being digested at all, use {X}|{X}. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--digestion full-digest|partial-digest|non-specific-digest</code> &ndash; Specify whether every peptide in the database must have two enzymatic termini (full-digest) or if peptides with only one enzymatic terminus are also included (partial-digest). Default = <code>full-digest</code>.</li>
  <li class="nobullet"><code>--enzyme no-enzyme|trypsin|trypsin/p|chymotrypsin|elastase|clostripain|cyanogen-bromide|iodosobenzoate|proline-endopeptidase|staph-protease|asp-n|lys-c|lys-n|arg-c|glu-c|pepsin-a|elastase-trypsin-chymotrypsin|lysarginase|custom-enzyme</code> &ndash; Specify the enzyme used to digest the proteins in silico. Available enzymes (with the corresponding digestion rules indicated in parentheses) include no-enzyme ([X]|[X]), trypsin ([RK]|{P}), trypsin/p ([RK]|[]), chymotrypsin ([FWYL]|{P}), elastase ([ALIV]|{P}), clostripain ([R]|[]), cyanogen-bromide ([M]|[]), iodosobenzoate ([W]|[]), proline-endopeptidase ([P]|[]), staph-protease ([E]|[]), asp-n ([]|[D]), lys-c ([K]|{P}), lys-n ([]|[K]), arg-c ([R]|{P}), glu-c ([DE]|{P}), pepsin-a ([FL]|{P}), elastase-trypsin-chymotrypsin ([ALIVKRWFY]|{P}), lysarginase ([]|[KR]). Specifying --enzyme no-enzyme yields a non-enzymatic digest. <strong>Warning:</strong> the resulting index may be quite large. Default = <code>trypsin</code>.</li>
  <li class="nobullet"><code>--missed-cleavages &lt;integer&gt;</code> &ndash; Maximum number of missed cleavages per peptide to allow in enzymatic digestion. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Precursor selection</h3>
<ul>
  <li class="nobullet"><code>--auto-precursor-window false|warn|fail</code> &ndash; Automatically estimate optimal value for the precursor-window parameter from the spectra themselves. false=no estimation, warn=try to estimate but use the default value in case of failure, fail=try to estimate and quit in case of failure. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--max-precursor-charge &lt;integer&gt;</code> &ndash; The maximum charge state of a spectra to consider in search. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--min-precursor-charge &lt;integer&gt;</code> &ndash; The minimum charge state of a spectra to consider in search. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--precursor-window &lt;float&gt;</code> &ndash; Tolerance used for matching peptides to spectra. Peptides must be within +/- 'precursor-window' of the spectrum value. The precursor window units depend upon precursor-window-type. Default = <code>50</code>.</li>
  <li class="nobullet"><code>--precursor-window-type mass|mz|ppm</code> &ndash; Specify the units for the window that is used to select peptides around the precursor mass location (mass, mz, ppm). The magnitude of the window is defined by the precursor-window option, and candidate peptides must fall within this window. For the mass window-type, the spectrum precursor m+h value is converted to mass, and the window is defined as that mass +/- precursor-window. If the m+h value is not available, then the mass is calculated from the precursor m/z and provided charge. The peptide mass is computed as the sum of the monoisotopic amino acid masses plus 18 Da for the terminal OH group. The mz window-type calculates the window as spectrum precursor m/z +/- precursor-window and then converts the resulting m/z range to the peptide mass range using the precursor charge. For the parts-per-million (ppm) window-type, the spectrum mass is calculated as in the mass type. The lower bound of the mass window is then defined as the spectrum mass / (1.0 + (precursor-window / 1000000)) and the upper bound is defined as spectrum mass / (1.0 - (precursor-window / 1000000)). Default = <code>ppm</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Search parameters</h3>
<ul>
  <li class="nobullet"><code>--auto-mz-bin-width false|warn|fail</code> &ndash; Automatically estimate optimal value for the mz-bin-width parameter from the spectra themselves. false=no estimation, warn=try to estimate but use the default value in case of failure, fail=try to estimate and quit in case of failure. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--compute-sp T|F</code> &ndash; Compute the preliminary score Sp for all candidate peptides. Report this score in the output, along with the corresponding rank, the number of matched ions and the total number of ions. This option is recommended if results are to be analyzed by Percolator.If sqt-output is enabled, then compute-sp is automatically enabled and cannot be overridden. Note that the Sp computation requires re-processing each observed spectrum, so turning on this switch involves significant computational overhead. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--deisotope &lt;float&gt;</code> &ndash; Perform a simple deisotoping operation across each MS2 spectrum. For each peak in an MS2 spectrum, consider lower m/z peaks. If the current peak occurs where an expected peak would lie for any charge state less than the charge state of the precursor, within mass tolerance, and if the current peak is of lower abundance, then the peak is removed.  The value of this parameter is the mass tolerance, in units of parts-per-million.  If set to 0, no deisotoping is performed. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--exact-p-value T|F</code> &ndash; Enable the calculation of exact p-values for the XCorr score as described in <a href="http://www.ncbi.nlm.nih.gov/pubmed/24895379">this article</a>. Calculation of p-values increases the running time but increases the number of identifications at a fixed confidence threshold. The p-values will be reported in a new column with header "exact p-value", and the "xcorr score" column will be replaced with a "refactored xcorr" column. Note that, currently, p-values can only be computed when the mz-bin-width parameter is set to its default value. Variable and static mods are allowed on non-terminal residues in conjunction with p-value computation, but currently only static mods are allowed on the N-terminus, and no mods on the C-terminus. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--isotope-error &lt;string&gt;</code> &ndash; List of positive, non-zero integers. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--min-peaks &lt;integer&gt;</code> &ndash; The minimum number of peaks a spectrum must have for it to be searched. Default = <code>20</code>.</li>
  <li class="nobullet"><code>--mz-bin-offset &lt;float&gt;</code> &ndash; In the discretization of the m/z axes of the observed and theoretical spectra, this parameter specifies the location of the left edge of the first bin, relative to mass = 0 (i.e., mz-bin-offset = 0.xx means the left edge of the first bin will be located at +0.xx Da). Default = <code>0.4</code>.</li>
  <li class="nobullet"><code>--mz-bin-width &lt;float&gt;</code> &ndash; Before calculation of the XCorr score, the m/z axes of the observed and theoretical spectra are discretized. This parameter specifies the size of each bin. The exact formula for computing the discretized m/z value is floor((x/mz-bin-width) + 1.0 - mz-bin-offset), where x is the observed m/z value. For low resolution ion trap ms/ms data 1.0005079 and for high resolution ms/ms 0.02 is recommended. Default = <code>0.02</code>.</li>
  <li class="nobullet"><code>--peptide-centric-search T|F</code> &ndash; Carries out a peptide-centric search. For each peptide the top-scoring spectra are reported, in contrast to the standard spectrum-centric search where the top-scoring peptides are reported. Note that in this case the "xcorr rank" column will contain the rank of the given spectrum with respect to the given candidate peptide, rather than vice versa (which is the default). Default = <code>false</code>.</li>
  <li class="nobullet"><code>--score-function xcorr|residue-evidence|both</code> &ndash; Function used for scoring PSMs. 'xcorr' is the original scoring function used by SEQUEST; 'residue-evidence' is designed to score high-resolution MS2 spectra; and 'both' calculates both scores. The latter requires that exact-p-value=T. Default = <code>xcorr</code>.</li>
  <li class="nobullet"><code>--fragment-tolerance &lt;float&gt;</code> &ndash; Mass tolerance (in Da) for scoring pairs of peaks when creating the residue evidence matrix. This parameter only makes sense when score-function is 'residue-evidence' or 'both'. Default = <code>0.02</code>.</li>
  <li class="nobullet"><code>--evidence-granularity &lt;integer&gt;</code> &ndash; This parameter controls the granularity of the entries in the dynamic programming matrix used in residue-evidence scoring.Smaller values make the program run faster but give less accurate p-values; larger values make the program run more slowly but give more accurate p-values. Default = <code>25</code>.</li>
  <li class="nobullet"><code>--remove-precursor-peak T|F</code> &ndash; If true, all peaks around the precursor m/z will be removed, within a range specified by the --remove-precursor-tolerance option. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--remove-precursor-tolerance &lt;float&gt;</code> &ndash; This parameter specifies the tolerance (in Th) around each precursor m/z that is removed when the --remove-precursor-peak option is invoked. Default = <code>1.5</code>.</li>
  <li class="nobullet"><code>--scan-number &lt;string&gt;</code> &ndash; A single scan number or a range of numbers to be searched. Range should be specified as 'first-last' which will include scans 'first' and 'last'. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--skip-preprocessing T|F</code> &ndash; Skip preprocessing steps on spectra. Default = F. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--spectrum-max-mz &lt;float&gt;</code> &ndash; The highest spectrum m/z to search in the ms2 file. Default = <code>1e+09</code>.</li>
  <li class="nobullet"><code>--spectrum-min-mz &lt;float&gt;</code> &ndash; The lowest spectrum m/z to search in the ms2 file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use-flanking-peaks T|F</code> &ndash; Include flanking peaks around singly charged b and y theoretical ions. Each flanking peak occurs in the adjacent m/z bin and has half the intensity of the primary peak. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--use-neutral-loss-peaks T|F</code> &ndash; Controls whether neutral loss ions are considered in the search. For XCorr, the loss of ammonia (NH3, 17.0086343 Da) is applied to singly charged b- and y-ions, and the loss of water (H2O; 18.0091422) is applied to b-ions. If the precursor charge is >=3, then a doubly-charged version of each ion is added. For XCorr p-value, three types of neutral losses are included. Loss of ammonia and water are applied to b- and y-ions, and a carbon monoxide loss (CO, 27.9949) is also applied to b-ions. Higher charge fragments are included for all possible charges less than the precursor charge. All neutral loss peaks have an intensity 1/10 of the primary peak. Neutral losses are not yet implemented for the res-ev score function. Default = <code>true</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Protein inference options</h3>
<ul>
  <li class="nobullet"><code>--picked-protein &lt;string&gt;</code> &ndash; Use the picked protein-level FDR to infer protein probabilities, provide the fasta file as the argument to this flag. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--protein-enzyme no_enzyme|elastase|pepsin|proteinasek|thermolysin|trypsinp|chymotrypsin|lys-n|lys-c|arg-c|asp-n|glu-c|lysarginase|trypsin</code> &ndash; Type of enzyme Default = <code>trypsin</code>.</li>
  <li class="nobullet"><code>--protein-report-duplicates T|F</code> &ndash; If multiple database proteins contain exactly the same set of peptides, then Percolator will randomly discard all but one of the proteins. If this option is set, then the IDs of these duplicated proteins will be reported as a comma-separated list. Not available for Fido. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--protein-report-fragments T|F</code> &ndash; By default, if the peptides associated with protein A are a proper subset of the peptides associated with protein B, then protein A is eliminated and all the peptides are considered as evidence for protein B. Note that this filtering is done based on the complete set of peptides in the database, not based on the identified peptides in the search results. Alternatively, if this option is set and if all of the identified peptides associated with protein B are also associated with protein A, then Percolator will report a comma-separated list of protein IDs, where the full-length protein B is first in the list and the fragment protein A is listed second. Not available for Fido. Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Fido options</h3>
<ul>
  <li class="nobullet"><code>--fido-alpha &lt;float&gt;</code> &ndash; Specify the probability with which a present protein emits an associated peptide. Set by grid search (see --fido-gridsearch-depth parameter) if not specified. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fido-beta &lt;float&gt;</code> &ndash; Specify the probability of the creation of a peptide from noise. Set by grid search (see --fido-gridsearch-depth parameter) if not specified. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fido-empirical-protein-q T|F</code> &ndash; Estimate empirical p-values and q-values for proteins using target-decoy analysis. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--fido-fast-gridsearch &lt;float&gt;</code> &ndash; Apply the specified threshold to PSM, peptide and protein probabilities to obtain a faster estimate of the alpha, beta and gamma parameters. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fido-gamma &lt;float&gt;</code> &ndash; Specify the prior probability that a protein is present in the sample. Set by grid search (see --fido-gridsearch-depth parameter) if not specified. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fido-gridsearch-depth &lt;integer&gt;</code> &ndash; Set depth of the grid search for alpha, beta and gamma estimation. The values considered, for each possible value of the --fido-gridsearch-depth parameter, are as follows:<ul><li>0: alpha = {0.01, 0.04, 0.09, 0.16, 0.25, 0.36, 0.5}; beta = {0.0, 0.01, 0.15, 0.025, 0.035, 0.05, 0.1}; gamma = {0.1, 0.25, 0.5, 0.75}.</li><li>1: alpha = {0.01, 0.04, 0.09, 0.16, 0.25, 0.36}; beta = {0.0, 0.01, 0.15, 0.025, 0.035, 0.05}; gamma = {0.1, 0.25, 0.5}.</li><li>2: alpha = {0.01, 0.04, 0.16, 0.25, 0.36}; beta = {0.0, 0.01, 0.15, 0.030, 0.05}; gamma = {0.1, 0.5}.</li><li>3: alpha = {0.01, 0.04, 0.16, 0.25, 0.36}; beta = {0.0, 0.01, 0.15, 0.030, 0.05}; gamma = {0.5}.</li></ul> Default = <code>0</code>.</li>
  <li class="nobullet"><code>--fido-gridsearch-mse-threshold &lt;float&gt;</code> &ndash; Q-value threshold that will be used in the computation of the MSE and ROC AUC score in the grid search. Default = <code>0.05</code>.</li>
  <li class="nobullet"><code>--fido-no-split-large-components T|F</code> &ndash; Do not approximate the posterior distribution by allowing large graph components to be split into subgraphs. The splitting is done by duplicating peptides with low probabilities. Splitting continues until the number of possible configurations of each subgraph is below 2^18 Default = <code>false</code>.</li>
  <li class="nobullet"><code>--fido-protein-truncation-threshold &lt;float&gt;</code> &ndash; To speed up inference, proteins for which none of the associated peptides has a probability exceeding the specified threshold will be assigned probability = 0. Default = <code>0.01</code>.</li>
  <li class="nobullet"><code>--protein T|F</code> &ndash; Use the Fido algorithm to infer protein probabilities. Must be true to use any of the Fido options. Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Database</h3>
<ul>
  <li class="nobullet"><code>--decoy_search &lt;integer&gt;</code> &ndash; 0=no, 1=concatenated search, 2=separate search. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--peff_format &lt;integer&gt;</code> &ndash; 0=normal FASTA format,1=PEFF PSI-MOD modifications and amino acid variants,2=PEFF Unimod modifications and amino acid variants,3=PEFF PSI-MOD modifications, skipping amino acid variants,4=PEFF Unimod modifications, skipping amino acid variants,5=PEFF amino acid variants, skipping PEFF modifications. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--peff_obo &lt;string&gt;</code> &ndash; A full or relative path to the OBO file used with a PEFF search. Supported OBO formats are PSI-Mod and Unimod OBO files. Which OBO file you use depends on your PEFF input file. This parameter is ignored if "peff_format = 0". There is no default value if this parameter is missing. Default = <code>&lt;empty&gt;</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>CPU threads</h3>
<ul>
  <li class="nobullet"><code>--num-threads &lt;integer&gt;</code> &ndash; 0=poll CPU to set num threads; else specify num threads directly. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--num_threads &lt;integer&gt;</code> &ndash; 0=poll CPU to set num threads; else specify num threads directly. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Masses</h3>
<ul>
  <li class="nobullet"><code>--peptide_mass_tolerance &lt;float&gt;</code> &ndash; Controls the mass tolerance value.  The mass tolerance is set at +/- the specified number i.e. an entered value of "1.0" applies a -1.0 to +1.0 tolerance. The units of the mass tolerance is controlled by the parameter "peptide_mass_units".  Default = <code>3</code>.</li>
  <li class="nobullet"><code>--auto_peptide_mass_tolerance false|warn|fail</code> &ndash; Automatically estimate optimal value for the peptide_mass_tolerance parameter from the spectra themselves. false=no estimation, warn=try to estimate but use the default value in case of failure, fail=try to estimate and quit in case of failure. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--peptide_mass_units &lt;integer&gt;</code> &ndash; 0=amu, 1=mmu, 2=ppm. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--mass_type_parent &lt;integer&gt;</code> &ndash; 0=average masses, 1=monoisotopic masses. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--mass_type_fragment &lt;integer&gt;</code> &ndash; 0=average masses, 1=monoisotopic masses. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--precursor_tolerance_type &lt;integer&gt;</code> &ndash; 0=singly charged peptide mass, 1=precursor m/z. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--isotope_error &lt;integer&gt;</code> &ndash; 0=off, 1=0/1 (C13 error), 2=0/1/2, 3=0/1/2/3, 4=--8/-4/0/4/8 (for +4/+8 labeling), 5=-1/0/1/2/3. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Search enzyme</h3>
<ul>
  <li class="nobullet"><code>--search_enzyme_number &lt;integer&gt;</code> &ndash; Specify a search enzyme from the end of the parameter file. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--search_enzyme2_number &lt;integer&gt;</code> &ndash; Specify a second search enzyme from the end of the parameter file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--num_enzyme_termini &lt;integer&gt;</code> &ndash; valid values are 1 (semi-digested), 2 (fully digested), 8 N-term, 9 C-term. Default = <code>2</code>.</li>
  <li class="nobullet"><code>--allowed_missed_cleavage &lt;integer&gt;</code> &ndash; Maximum value is 5; for enzyme search. Default = <code>2</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Fragment ions</h3>
<ul>
  <li class="nobullet"><code>--fragment_bin_tol &lt;float&gt;</code> &ndash; Binning to use on fragment ions. Default = <code>1.000507</code>.</li>
  <li class="nobullet"><code>--fragment_bin_offset &lt;float&gt;</code> &ndash; Offset position to start the binning (0.0 to 1.0). Default = <code>0.4</code>.</li>
  <li class="nobullet"><code>--auto_fragment_bin_tol false|warn|fail</code> &ndash; Automatically estimate optimal value for the fragment_bin_tol parameter from the spectra themselves. false=no estimation, warn=try to estimate but use the default value in case of failure, fail=try to estimate and quit in case of failure. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--theoretical_fragment_ions &lt;integer&gt;</code> &ndash; 0=default peak shape, 1=M peak only. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--use_A_ions &lt;integer&gt;</code> &ndash; Controls whether or not A-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use_B_ions &lt;integer&gt;</code> &ndash; Controls whether or not B-ions are considered in the search (0 - no, 1 - yes). Default = <code>1</code>.</li>
  <li class="nobullet"><code>--use_C_ions &lt;integer&gt;</code> &ndash; Controls whether or not C-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use_X_ions &lt;integer&gt;</code> &ndash; Controls whether or not X-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use_Y_ions &lt;integer&gt;</code> &ndash; Controls whether or not Y-ions are considered in the search (0 - no, 1 - yes). Default = <code>1</code>.</li>
  <li class="nobullet"><code>--use_Z_ions &lt;integer&gt;</code> &ndash; Controls whether or not Z-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use_Z1_ions &lt;integer&gt;</code> &ndash; Controls whether or not Z1-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--use_NL_ions &lt;integer&gt;</code> &ndash; 0=no, 1= yes to consider NH3/H2O neutral loss peak. Default = <code>1</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>mzXML/mzML parameters</h3>
<ul>
  <li class="nobullet"><code>--scan_range &lt;string&gt;</code> &ndash; Start and scan scan range to search; 0 as first entry ignores parameter. Default = <code>0 0</code>.</li>
  <li class="nobullet"><code>--precursor_charge &lt;string&gt;</code> &ndash; Precursor charge range to analyze; does not override mzXML charge; 0 as first entry ignores parameter. Default = <code>0 0</code>.</li>
  <li class="nobullet"><code>--override_charge &lt;integer&gt;</code> &ndash; Specifies the whether to override existing precursor charge state information when present in the files with the charge range specified by the "precursor_charge" parameter. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--ms_level &lt;integer&gt;</code> &ndash; MS level to analyze, valid are levels 2 or 3. Default = <code>2</code>.</li>
  <li class="nobullet"><code>--activation_method ALL|CID|ECD|ETD+SA|ETD|PQD|HCD|IRMPD</code> &ndash; Specifies which scan types are searched. Default = <code>ALL</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Miscellaneous parameters</h3>
<ul>
  <li class="nobullet"><code>--clip_nterm_methionine &lt;integer&gt;</code> &ndash; 0=leave sequences as-is; 1=also consider sequence w/o N-term methionine. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--decoy_prefix &lt;string&gt;</code> &ndash; Specifies the prefix of the protein names that indicates a decoy. Default = <code>decoy_</code>.</li>
  <li class="nobullet"><code>--digest_mass_range &lt;string&gt;</code> &ndash; MH+ peptide mass range to analyze. Default = <code>600.0 5000.0</code>.</li>
  <li class="nobullet"><code>--equal_I_and_L &lt;integer&gt;</code> &ndash; This parameter controls whether the Comet treats isoleucine (I) and leucine (L) as the same/equivalent with respect to a peptide identification. 0 treats I and L as different, 1 treats I and L as the same. The default value is "1" Default = <code>1</code>.</li>
  <li class="nobullet"><code>--mass_offsets &lt;string&gt;</code> &ndash; Specifies one or more mass offsets to apply. This value(s) are effectively subtracted from each precursor mass such that peptides that are smaller than the precursor mass by the offset value can still be matched to the respective spectrum. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--max_duplicate_proteins &lt;integer&gt;</code> &ndash; defines the maximum number of proteins (identifiers/accessions) to report. If a peptide is present in 6 total protein sequences, there is one (first) reference protein and 5 additional duplicate proteins. This parameter controls how many of those 5 additional duplicate proteins are reported.If "decoy_search = 2" is set to report separate target and decoy results, this parameter will be applied to the target and decoy outputs separately. If set to "-1", there will be no limit on the number of reported additional proteins. The default value is "20" if this parameter is missing. Default = <code>20</code>.</li>
  <li class="nobullet"><code>--max_fragment_charge &lt;integer&gt;</code> &ndash; Set maximum fragment charge state to analyze (allowed max 5). Default = <code>3</code>.</li>
  <li class="nobullet"><code>--max_index_runtime &lt;integer&gt;</code> &ndash; Sets the maximum indexed database search run time for a scan/query. Valid values are integers 0 or higher representing the maximum run time in milliseconds. As Comet loops through analyzing peptides from the database index file, it checks the cummulative run time of that spectrum search after each peptide is analyzed. If the run time exceeds the value set for this parameter, the search is aborted and the best peptide result analyzed up to that point is returned. To have no maximum search time, set this parameter value to "0". The default value is "0". Default = <code>0</code>.</li>
  <li class="nobullet"><code>--max_precursor_charge &lt;integer&gt;</code> &ndash; Set maximum precursor charge state to analyze (allowed max 9). Default = <code>6</code>.</li>
  <li class="nobullet"><code>--num_results &lt;integer&gt;</code> &ndash; Number of search hits to store internally. Default = <code>50</code>.</li>
  <li class="nobullet"><code>--nucleotide_reading_frame &lt;integer&gt;</code> &ndash; 0=proteinDB, 1-6, 7=forward three, 8=reverse three, 9=all six. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--output_suffix &lt;string&gt;</code> &ndash; Specifies the suffix string that is appended to the base output name for the pep.xml, pin.xml, txt and sqt output files. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--peptide_length_range &lt;string&gt;</code> &ndash; Defines the length range of peptides to search. This parameter has two integer values. The first value is the minimum length cutoff and the second value is the maximum length cutoff. Only peptides within the specified length range are analyzed. The maximum peptide length that Comet can analyze is 63. The default values are "1 63". Default = <code>1 63</code>.</li>
  <li class="nobullet"><code>--precursor_NL_ions &lt;string&gt;</code> &ndash; Controls whether or not precursor neutral loss peaks are considered in the xcorr scoring. If left blank, this parameter is ignored.  To consider precursor neutral loss peaks, add one or more neutral loss mass value separated by a space.  Each entered mass value will be subtracted from the experimentral precursor mass and resulting neutral loss m/z values for all charge states (from 1 to precursor charge) will be analyzed. As these neutral loss peaks are analyzed along side fragment ion peaks, the fragment tolerance settings (fragment_bin_tol, fragment_bin_offset, theoretical_fragment_ion) apply to the precursor neutral loss peaks. The default value is blank/unused. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--skip_researching &lt;integer&gt;</code> &ndash; For '.out' file output only, 0=search everything again, 1=don't search if .out exists. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--spectrum_batch_size &lt;integer&gt;</code> &ndash; Maximum number of spectra to search at a time; 0 to search the entire scan range in one loop. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--text_file_extension &lt;string&gt;</code> &ndash; Specifies the a custom extension for output text file. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--explicit_deltacn &lt;integer&gt;</code> &ndash; 0=Comet deltaCn reported between the top peptide and the first dissimilar peptide, 1=Comet deltaCn reported between the top two peptides. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--old_mods_encoding &lt;integer&gt;</code> &ndash; 0=Comet will use mass based modification encodings, 1=Comet will use the old character based modification encodings. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Spectral processing</h3>
<ul>
  <li class="nobullet"><code>--minimum_peaks &lt;integer&gt;</code> &ndash; Minimum number of peaks in spectrum to search. Default = <code>10</code>.</li>
  <li class="nobullet"><code>--minimum_intensity &lt;float&gt;</code> &ndash; Minimum intensity value to read in. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--remove_precursor_peak &lt;integer&gt;</code> &ndash; 0=no, 1=yes, 2=all charge reduced precursor peaks (for ETD). Default = <code>0</code>.</li>
  <li class="nobullet"><code>--remove_precursor_tolerance &lt;float&gt;</code> &ndash; +- Da tolerance for precursor removal. Default = <code>1.5</code>.</li>
  <li class="nobullet"><code>--clear_mz_range &lt;string&gt;</code> &ndash; For iTRAQ/TMT type data; will clear out all peaks in the specified m/z range. Default = <code>0.0 0.0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Variable modifications</h3>
<ul>
  <li class="nobullet"><code>--variable_mod01 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported. Each modification is specified using seven entries: "&lt;mass&gt; &lt;residues&gt; &lt;type&gt; &lt;max&gt; &lt;distance&gt; &lt;terminus&gt; &lt;force&gt;." Type is 0 for static mods and non-zero for variable mods. Note that that if you set the same type value on multiple modification entries, Comet will treat those variable modifications as a binary set. This means that all modifiable residues in the binary set must be unmodified or modified. Multiple binary sets can be specified by setting a different binary modification value. Max is an integer specifying the maximum number of modified residues possible in a peptide for this modification entry. Distance specifies the distance the modification is applied to from the respective terminus: -1 = no distance contraint; 0 = only applies to terminal residue; N = only applies to terminal residue through next N residues. Terminus specifies which terminus the distance constraint is applied to: 0 = protein N-terminus; 1 = protein C-terminus; 2 = peptide N-terminus; 3 = peptide C-terminus.Force specifies whether peptides must contain this modification: 0 = not forced to be present; 1 = modification is required. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod02 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod03 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod04 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod05 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod06 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod07 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod08 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--variable_mod09 &lt;string&gt;</code> &ndash; See syntax for variable_mod01. Default = <code>0.0 null 0 4 -1 0 0</code>.</li>
  <li class="nobullet"><code>--auto_modifications T|F</code> &ndash; Automatically infer modifications from the spectra themselves. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--max_variable_mods_in_peptide &lt;integer&gt;</code> &ndash; Specifies the total/maximum number of residues that can be modified in a peptide. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--require_variable_mod &lt;integer&gt;</code> &ndash; Controls whether the analyzed peptides must contain at least one variable modification. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Static modifications</h3>
<ul>
  <li class="nobullet"><code>--add_Cterm_peptide &lt;float&gt;</code> &ndash; Specifiy a static modification to the c-terminus of all peptides. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Nterm_peptide &lt;float&gt;</code> &ndash; Specify a static modification to the n-terminus of all peptides. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Cterm_protein &lt;float&gt;</code> &ndash; Specify a static modification to the c-terminal peptide of each protein. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Nterm_protein &lt;float&gt;</code> &ndash; Specify a static modification to the n-terminal peptide of each protein. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_A_alanine &lt;float&gt;</code> &ndash; Specify a static modification to the residue A. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_C_cysteine &lt;float&gt;</code> &ndash; Specify a static modification to the residue C. Default = <code>57.021464</code>.</li>
  <li class="nobullet"><code>--add_D_aspartic_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue D. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_E_glutamic_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue E. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_F_phenylalanine &lt;float&gt;</code> &ndash; Specify a static modification to the residue F. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_G_glycine &lt;float&gt;</code> &ndash; Specify a static modification to the residue G. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_H_histidine &lt;float&gt;</code> &ndash; Specify a static modification to the residue H. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_I_isoleucine &lt;float&gt;</code> &ndash; Specify a static modification to the residue I. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_K_lysine &lt;float&gt;</code> &ndash; Specify a static modification to the residue K. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_L_leucine &lt;float&gt;</code> &ndash; Specify a static modification to the residue L. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_M_methionine &lt;float&gt;</code> &ndash; Specify a static modification to the residue M. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_N_asparagine &lt;float&gt;</code> &ndash; Specify a static modification to the residue N. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_O_pyrrolysine &lt;float&gt;</code> &ndash; Specify a static modification to the residue O. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_P_proline &lt;float&gt;</code> &ndash; Specify a static modification to the residue P. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Q_glutamine &lt;float&gt;</code> &ndash; Specify a static modification to the residue Q. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_R_arginine &lt;float&gt;</code> &ndash; Specify a static modification to the residue R. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_S_serine &lt;float&gt;</code> &ndash; Specify a static modification to the residue S. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_T_threonine &lt;float&gt;</code> &ndash; Specify a static modification to the residue T. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_U_selenocysteine &lt;float&gt;</code> &ndash; Specify a static modification to the residue U. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_V_valine &lt;float&gt;</code> &ndash; Specify a static modification to the residue V. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_W_tryptophan &lt;float&gt;</code> &ndash; Specify a static modification to the residue W. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Y_tyrosine &lt;float&gt;</code> &ndash; Specify a static modification to the residue Y. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_B_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue B. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_J_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue J. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_X_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue X. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--add_Z_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue Z. Default = <code>0</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>General options</h3>
<ul>
  <li class="nobullet"><code>--only-psms T|F</code> &ndash; Report results only at the PSM level.  This flag causes Percolator to skip the step that selects the top-scoring PSM per peptide; hence, peptide-level results are left out and only PSM-level results are reported. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--search-input auto|separate|concatenated</code> &ndash; Specify the type of target-decoy search. Using 'auto', percolator attempts to detect the search type automatically.  Using 'separate' specifies two searches: one against target and one against decoy protein db. Using 'concatenated' specifies a single search on concatenated target-decoy protein db. Default = <code>auto</code>.</li>
  <li class="nobullet"><code>--tdc T|F</code> &ndash; Use target-decoy competition to assign q-values and PEPs. When set to F, the mix-max method, which estimates the proportion pi0 of incorrect target PSMs, is used instead. Default = <code>true</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>SVM training options</h3>
<ul>
  <li class="nobullet"><code>--c-neg &lt;float&gt;</code> &ndash; Penalty for mistake made on negative examples. If not specified, then this value is set by cross validation over {0.1, 1, 10}. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--c-pos &lt;float&gt;</code> &ndash; Penalty for mistakes made on positive examples. If this value is set to 0, then it is set via cross validation over the values {0.1, 1, 10}, selecting the value that yields the largest number of PSMs identified at the q-value threshold set via the --test-fdr parameter. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--maxiter &lt;integer&gt;</code> &ndash; Maximum number of iterations for training. Default = <code>10</code>.</li>
  <li class="nobullet"><code>--percolator-seed &lt;string&gt;</code> &ndash; When given a unsigned integer value seeds the random number generator with that value. When given the string "time" seeds the random number generator with the system time. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--quick-validation T|F</code> &ndash; Quicker execution by reduced internal cross-validation. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--static T|F</code> &ndash; Use the provided initial weights as a static model. If used, the --init-weights option must be specified. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--subset-max-train &lt;integer&gt;</code> &ndash; Only train Percolator on a subset of PSMs, and use the resulting score vector to evaluate the other PSMs. Recommended when analyzing huge numbers (>1 million) of PSMs. When set to 0, all PSMs are used for training as normal. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--test-each-iteration T|F</code> &ndash; Measure performance on test set each iteration. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--test-fdr &lt;float&gt;</code> &ndash; False discovery rate threshold used in selecting hyperparameters during internal cross-validation and for reporting the final results. Default = <code>0.01</code>.</li>
  <li class="nobullet"><code>--train-fdr &lt;float&gt;</code> &ndash; False discovery rate threshold to define positive examples in training. Default = <code>0.01</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>SVM feature input options</h3>
<ul>
  <li class="nobullet"><code>--default-direction &lt;string&gt;</code> &ndash; In its initial round of training, Percolator uses one feature to induce a ranking of PSMs. By default, Percolator will select the feature that produces the largest set of target PSMs at a specified FDR threshold (cf. --train-fdr). This option allows the user to specify which feature is used for the initial ranking, using the name as a string from <a href="../file-formats/features.html">this table</a>. The name can be preceded by a hyphen (e.g. "-XCorr") to indicate that a lower value is better. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--init-weights &lt;string&gt;</code> &ndash; Read the unnormalized initial weights from the third line of the given file. This can be the output of the --output-weights option from a previous Percolator analysis. Note that the weights must be in the same order as features in the PSM input file(s) Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--klammer T|F</code> &ndash; Use retention time features calculated as in "Improving tandem mass spectrum identification using peptide retention time prediction across diverse chromatography conditions" by Klammer AA, Yi X, MacCoss MJ and Noble WS. (<em>Analytical Chemistry</em>. 2007 Aug 15;79(16):6111-8.). Default = <code>false</code>.</li>
  <li class="nobullet"><code>--output-weights T|F</code> &ndash; Output final weights to a file named "percolator.weights.txt". Default = <code>false</code>.</li>
  <li class="nobullet"><code>--override T|F</code> &ndash; By default, Percolator will examine the learned weights for each feature, and if the weight appears to be problematic, then percolator will discard the learned weights and instead employ a previously trained, static score vector. This switch allows this error checking to be overriden. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--unitnorm T|F</code> &ndash; Use unit normalization (i.e., linearly rescale each PSM's feature vector to have a Euclidean length of 1), instead of standard deviation normalization. Default = <code>false</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>param-medic options</h3>
<ul>
  <li class="nobullet"><code>--pm-charges &lt;string&gt;</code> &ndash; Precursor charge states to consider MS/MS spectra from, in measurement error estimation, provided as comma-separated values. Default = <code>0,2,3,4</code>.</li>
  <li class="nobullet"><code>--pm-max-frag-mz &lt;float&gt;</code> &ndash; Maximum fragment m/z value to use in measurement error estimation. Default = <code>1800</code>.</li>
  <li class="nobullet"><code>--pm-max-precursor-delta-ppm &lt;float&gt;</code> &ndash; Maximum ppm distance between precursor m/z values to consider two scans potentially generated by the same peptide for measurement error estimation. Default = <code>50</code>.</li>
  <li class="nobullet"><code>--pm-max-precursor-mz &lt;float&gt;</code> &ndash; Minimum precursor m/z value to use in measurement error estimation. Default = <code>1800</code>.</li>
  <li class="nobullet"><code>--pm-max-scan-separation &lt;integer&gt;</code> &ndash; Maximum number of scans two spectra can be separated by in order to be considered potentially generated by the same peptide, for measurement error estimation. Default = <code>1000</code>.</li>
  <li class="nobullet"><code>--pm-min-common-frag-peaks &lt;integer&gt;</code> &ndash; Number of the most-intense peaks that two spectra must share in order to potentially be generated by the same peptide, for measurement error estimation. Default = <code>20</code>.</li>
  <li class="nobullet"><code>--pm-min-frag-mz &lt;float&gt;</code> &ndash; Minimum fragment m/z value to use in measurement error estimation. Default = <code>150</code>.</li>
  <li class="nobullet"><code>--pm-min-peak-pairs &lt;integer&gt;</code> &ndash; Minimum number of peak pairs (for precursor or fragment) that must be successfully paired in order to attempt to estimate measurement error distribution. Default = <code>200</code>.</li>
  <li class="nobullet"><code>--pm-min-precursor-mz &lt;float&gt;</code> &ndash; Minimum precursor m/z value to use in measurement error estimation. Default = <code>400</code>.</li>
  <li class="nobullet"><code>--pm-min-scan-frag-peaks &lt;integer&gt;</code> &ndash; Minimum fragment peaks an MS/MS scan must contain to be used in measurement error estimation. Default = <code>40</code>.</li>
  <li class="nobullet"><code>--pm-pair-top-n-frag-peaks &lt;integer&gt;</code> &ndash; Number of fragment peaks per spectrum pair to be used in fragment error estimation. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--pm-top-n-frag-peaks &lt;integer&gt;</code> &ndash; Number of most-intense fragment peaks to consider for measurement error estimation, per MS/MS spectrum. Default = <code>30</code>.</li>
</ul>
</li>
<li class="nobullet">
<h3>Input and output</h3>
<ul>
  <li class="nobullet"><code>--fileroot &lt;string&gt;</code> &ndash; The fileroot string will be added as a prefix to all output file names. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--output-dir &lt;string&gt;</code> &ndash; The name of the directory where output files will be created. Default = <code>crux-output</code>.</li>
  <li class="nobullet"><code>--overwrite T|F</code> &ndash; Replace existing files if true or fail when trying to overwrite a file if false. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--spectrum-format |ms2|bms2|cms2|mgf</code> &ndash; The format to write the output spectra to. If empty, the spectra will be output in the same format as the MS2 input. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--parameter-file &lt;string&gt;</code> &ndash; A file containing parameters.  See the <a href="../file-formats/parameter-file.html">parameter documentation</a> page for details. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--verbosity &lt;integer&gt;</code> &ndash; Specify the verbosity of the current processes. Each level prints the following messages, including all those at lower verbosity levels: 0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on the progress of execution, 40-more progress information, 50-debug info, 60-detailed debug info. Default = <code>30</code>.</li>
  <li class="nobullet"><code>--decoy-prefix &lt;string&gt;</code> &ndash; Specifies the prefix of the protein names that indicate a decoy. Default = <code>decoy_</code>.</li>
  <li class="nobullet"><code>--mass-precision &lt;integer&gt;</code> &ndash; Set the precision for masses and m/z written to sqt and text files. Default = <code>4</code>.</li>
  <li class="nobullet"><code>--peptide-list T|F</code> &ndash; Create in the output directory a text file listing of all the peptides in the database, along with their corresponding decoy peptides, neutral masses and proteins, one per line. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--temp-dir &lt;string&gt;</code> &ndash; The name of the directory where temporary files will be created. If this parameter is blank, then the system temporary directory will be used Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--concat T|F</code> &ndash; When set to T, target and decoy search results are reported in a single file, and only the top-scoring N matches (as specified via --top-match) are reported for each spectrum, irrespective of whether the matches involve target or decoy peptides. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--file-column T|F</code> &ndash; Include the file column in tab-delimited output. Default = <code>true</code>.</li>
  <li class="nobullet"><code>--mzid-output T|F</code> &ndash; Output an mzIdentML results file to the output directory. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--pepxml-output T|F</code> &ndash; Output a pepXML results file to the output directory. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--pin-output T|F</code> &ndash; Output a Percolator input (PIN) file to the output directory. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--precision &lt;integer&gt;</code> &ndash; Set the precision for scores written to sqt and text files. Default = <code>8</code>.</li>
  <li class="nobullet"><code>--print-search-progress &lt;integer&gt;</code> &ndash; Show search progress by printing every n spectra searched. Set to 0 to show no search progress. Default = <code>1000</code>.</li>
  <li class="nobullet"><code>--spectrum-parser pwiz|mstoolkit</code> &ndash; Specify the parser to use for reading in MS/MS spectra. The default, ProteoWizard parser can read the MS/MS file formats listed <a href="http://proteowizard.sourceforge.net/formats.shtml">here</a>. The alternative is <a href="../mstoolkit.html">MSToolkit parser</a>. If the ProteoWizard parser fails to read your files properly, you may want to try the MSToolkit parser instead. Default = <code>pwiz</code>.</li>
  <li class="nobullet"><code>--sqt-output T|F</code> &ndash; Outputs an SQT results file to the output directory. Note that if sqt-output is enabled, then compute-sp is automatically enabled and cannot be overridden. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--store-index &lt;string&gt;</code> &ndash; When providing a FASTA file as the index, the generated binary index will be stored at the given path. This option has no effect if a binary index is provided as the index. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--store-spectra &lt;string&gt;</code> &ndash; Specify the name of the file where the binarized fragmentation spectra will be stored. Subsequent runs of crux tide-search will execute more quickly if provided with the spectra in binary format. The filename is specified relative to the current working directory, not the Crux output directory (as specified by --output-dir). This option is not valid if multiple input spectrum files are given. Default = <code>&lt;empty&gt;</code>.</li>
  <li class="nobullet"><code>--top-match &lt;integer&gt;</code> &ndash; Specify the number of matches to report for each spectrum. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--txt-output T|F</code> &ndash; Output a tab-delimited results file to the output directory. Default = <code>true</code>.</li>
  <li class="nobullet"><code>--brief-output T|F</code> &ndash; Output in tab-delimited text only the file name, scan number, charge, score and peptide.Incompatible with mzid-output=T, pin-output=T, pepxml-output=T or txt-output=F. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--use-z-line T|F</code> &ndash; Specify whether, when parsing an MS2 spectrum file, Crux obtains the precursor mass information from the "S" line or the "Z" line.  Default = <code>true</code>.</li>
  <li class="nobullet"><code>--output_mzidentmlfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write mzIdentML file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--output_pepxmlfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write pep.xml file. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--output_percolatorfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes write percolator file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--output_sqtfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write sqt file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--output_sqtstream &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write sqt file. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--output_txtfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write tab-delimited text file. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--print_expect_score &lt;integer&gt;</code> &ndash; 0=no, 1=yes to replace Sp with expect in out & sqt. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--num_output_lines &lt;integer&gt;</code> &ndash; num peptide results to show. Default = <code>5</code>.</li>
  <li class="nobullet"><code>--show_fragment_ions &lt;integer&gt;</code> &ndash; 0=no, 1=yes for out files only. Default = <code>0</code>.</li>
  <li class="nobullet"><code>--sample_enzyme_number &lt;integer&gt;</code> &ndash; Sample enzyme which is possibly different than the one applied to the search. Used to calculate NTT & NMC in pepXML output. Default = <code>1</code>.</li>
  <li class="nobullet"><code>--decoy-xml-output T|F</code> &ndash; Include decoys (PSMs, peptides, and/or proteins) in the XML output. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--feature-file-out T|F</code> &ndash; Output the computed features in <a href="../file-formats/features.html">tab-delimited Percolator input (.pin) format</a>. The features will be normalized, using either unit norm or standard deviation normalization (depending upon the value of the unit-norm option). Default = <code>false</code>.</li>
  <li class="nobullet"><code>--pout-output T|F</code> &ndash; Output a Percolator <a href="https://github.com/percolator/percolator/blob/master/src/xml/percolator_out.xsd">pout.xml</a> format results file to the output directory. Default = <code>false</code>.</li>
  <li class="nobullet"><code>--list-of-files T|F</code> &ndash; Specify that the search results are provided as lists of files, rather than as individual files. Default = <code>false</code>.</li>
</ul>
</li>

</ul>
    <!-- END CONTENT -->
    </div>
  </div>
<footer class="site-footer">
  <div id="centerfooter">
    <div class="footerimportantlinks">
      <img src="../images/linkicon.png" style="width:16px; height:16px"><h3>Important links</h3>
      <ul>
        <li><a href="../faq.html">Crux FAQ</a></li>
        <li><a href="../glossary.html">Glossary of terminology</a></li>
        <li><a href="http://scholar.google.com/citations?hl=en&user=Rw9S1HIAAAAJ">Google Scholar profile</a></li>
        <li><a href="https://github.com/crux-toolkit/crux-toolkit">GitHub</a></li>
        <li><a href="../release-notes.html">Release Notes</a></li>
        <li><a href="https://mailman1.u.washington.edu/mailman/listinfo/crux-users" title="Receive announcements of new versions">Join the mailing list</a></li>
        <li><a href="http://www.apache.org/licenses/LICENSE-2.0">Apache license</a></li>
        <li><a href="http://groups.google.com/group/crux-users">Support Board</a></li>
      </ul>
    </div>
    <div class="footerimportantlinks tutoriallinks">
      <img src="../images/tutorialicon.png" style="height:16px"><h3>Tutorials</h3>
      <ul>
        <li><a href="../tutorials/install.html">Installation</a></li>
        <li><a href="../tutorials/gettingstarted.html">Getting started with Crux</a></li>
        <li><a href="../tutorials/search.html">Running a simple search using Tide and Percolator</a></li>
        <li><a href="../tutorials/customizedsearch.html">Customization and search options</a></li>
        <li><a href="../tutorials/spectralcounts.html">Using spectral-counts</a></li>
      </ul>
    </div>
    <div id="footertext">
      <p>
        The original version of Crux was written by Chris Park and Aaron Klammer
        under the supervision
        of <a href="http://www.gs.washington.edu/faculty/maccoss.htm">Prof. Michael
        MacCoss</a>
        and <a href="http://noble.gs.washington.edu/~noble">Prof. William
        Stafford Noble</a> in the Department of Genome Sciences at the
        University of Washington, Seattle.  Website by <a href="http://www.yuvalboss.com/">Yuval Boss</a>. 
        <br />The complete list of contributors
        can be found <a href="../contributors.html">here</a>.
        <br />
        <br />
        Maintenance and development of Crux is funded by the <a href="https://www.nih.gov/">National Institutes of Health</a> award P41 GM103533. 
      </p>
    </div>
  </div>
</footer>
</body>
</html>